# Bioconductor Data Structures {#BioC}

[Bioconductor](https://www.bioconductor.org/) promotes the statistical analysis and comprehension of current and emerging high-throughput biological assays. Bioconductor is based on packages written primarily in the R programming language. Bioconductor is committed to open source, collaborative, distributed software development and literate, reproducible research. Most Bioconductor components are distributed as [R](https://cran.r-project.org/) packages. The functional scope of [Bioconductor packages](http://bioconductor.org/packages/release/BiocViews.html#___Software) includes the analysis of DNA microarray, sequence, flow, SNP, and other data.

Bioconductor provides several data infrastructures for efficiently managing omic data. See [this paper](https://www.nature.com/articles/nmeth.3252) for a global overview.  Here we provide a quick introduction for the most commonly used ones. We also recommend to learn how to deal with [GenomicRanges](https://blog.dominodatalab.com/genomic-ranges-an-introduction-to-working-with-genomic-data/) which helps to efficiently manage genomic data information. 

## SNP Array Data

SNP array data are normally stored in [PLINK](http://zzz.bwh.harvard.edu/plink/) format (or [VCF](https://en.wikipedia.org/wiki/Variant_Call_Format) for NGS data). PLINK data are normally stored in three files .ped, .bim, .fam. The advantage is that SNP data are stored in binary format in the BED file (Homozygous normal 01, Heterozygous 02, Homozygous variant 03, missing 00). 

- FAM  file:  one row per individual - identification information: Family ID, Individual ID
Paternal ID, Maternal ID, Sex (1=male; 2=female; other=unknown), Phenotype.
- BIM file:  one row per SNP (rs id, chromosome, position, allele 1, allele 2).
- BED  file:  one row per individual. Genotypes in columns.

Data are easily loaded into R by using `read.plink` function

```{r, read_plink}
require(snpStats)
snps <- read.plink("data/obesity") # there are three files obesity.fam, obesity.bim, obesity.bed
names(snps)
```

Genotypes is a `snpMatrix` object

```{r, genotypes}
geno <- snps$genotypes
geno
```

Annotation is a `data.frame` object

```{r, annotation}
annotation <- snps$map
head(annotation)
```

## Expression Sets

[ExpressionSet]()

![Alt ExpressionSet](fig/eSet_vs_dataframe.png)

Description

- `Biobase` is part of the Bioconductor project and contains standardized data structures to represent genomic data. 

- The `ExpressionSet` class is designed to combine several different sources of information into a single convenient structure. 

- An `ExpressionSet` can be manipulated (e.g., subsetted, copied) conveniently, and is the input or output from many Bioconductor functions.

- The data in an `ExpressionSet` consists of expression data from microarray experiments, `meta-data' describing samples in the experiment, annotations and meta-data about the features on the chip and information related to the protocol used for processing each sample 

- Print

```{r, getExpressionSet}
library(tweeDEseqCountData)
data(pickrell)
pickrell.eset
```

- Get experimental data (e.g., gene expression)

```{r, getGeneExpression}
genes <- exprs(pickrell.eset)
genes[1:4,1:4]
```

Get phenotypic data (e.g. covariates, disease status, outcomes, ...)

```{r, getPheno}
pheno <- pData(pickrell.eset)
head(pheno)
pheno$gender
```

This also works

```{r, getPheno2}
pickrell.eset$gender
```

Subsetting (everything is synchronized)

```{r, subset}
eSet.male <- pickrell.eset[, pickrell.eset$gender=="male"]
eSet.male
```

Finally, the `fData` function gets the probes' annotation in a `data.frame`. Let us first illustrate how to provide an annotation to an `ExpressionSet` object

```{r add_annot_eSet}
require(Homo.sapiens)
geneSymbols <- rownames(genes)
annot <- select(Homo.sapiens, geneSymbols, 
                columns=c("TXCHROM", "SYMBOL"), keytype="ENSEMBL")
annotation(pickrell.eset) <- "Homo.sapiens"
fData(pickrell.eset) <- annot
pickrell.eset
```

```{r getProbes} 
probes <- fData(pickrell.eset)
head(probes)
```

## Genomic Ranges

[GenomicRanges]()

- `GRanges()`: genomic coordinates to represent annotations (exons,
  genes, regulatory marks, ...) and data (called peaks, variants,
  aligned reads)

- `GRangesList()`: genomic coordinates grouped into list elements
  (e.g., paired-end reads; exons grouped by transcript)


Operations

- intra-range: act on each range independently
    - e.g., `shift()`
- inter-range: act on all ranges in a `GRanges` object or
  `GRangesList` element
      - e.g., `reduce()`; `disjoin()`
- between-range: act on two separate `GRanges` or `GRangesList`
  objects
      - e.g., `findOverlaps()`, `nearest()`

```{r ranges, message=FALSE}
gr <- GRanges("chr1", IRanges(c(10, 20, 22), width=5), "+")
gr

# shift move all intervals 3 base pair towards the end
shift(gr, 3) 

# inter-range
range(gr)    

# two Granges: knowing the intervals that overlap with a targeted region
snps <- GRanges("chr1", IRanges(c(11, 17), width=1))
snps
gr.ov <- findOverlaps(snps, gr)
gr.ov

# recover the overlapping intervals
gr[subjectHits(gr.ov)]

#coverage: summarizes the times each base is covered by an interval
coverage(gr)

# get counts
countOverlaps(gr, snps)
```

This table shows the common operations of `GenomicRanges`

\begin{table}
\footnotesize
  \centering
  \caption{Common operations on \Rclass{IRanges}, \Rclass{GRanges} and
    \Rclass{GRangesList}. Table obtained from \url{https://bioconductor.org/help/course-materials/2014/SeattleFeb2014/}.}
  \label{tab:range-ops}
  \begin{tabular}{lll}
    Category & Function & Description \\
    \hline

    Accessors & \Rfunction{start}, \Rfunction{end}, \Rfunction{width} & 
        Get or set the starts, ends and widths \\
    & \Rfunction{names} & Get or set the names \\
    & \Rfunction{mcols}, \Rfunction{metadata} & 
        Get or set metadata on elements or object \\
    & \Rfunction{length} & Number of ranges in the vector \\
    & \Rfunction{range} & Range formed from min start and max end \\

    Ordering & \Rfunction{<}, \Rfunction{<=}, \Rfunction{>}, \Rfunction{>=}, 
        \Rfunction{==}, \Rfunction{!=} & 
        Compare ranges, ordering by start then width \\
    & \Rfunction{sort}, \Rfunction{order}, \Rfunction{rank} & 
        Sort by the ordering \\
    & \Rfunction{duplicated} & Find ranges with multiple instances \\
    & \Rfunction{unique} & Find unique instances, removing duplicates \\

    Arithmetic & \Robject{r + x}, \Robject{r - x}, \Robject{r * x} & 
        Shrink or expand ranges \Robject{r} by number \Robject{x} \\
    & \Rfunction{shift} & Move the ranges by specified amount \\
    & \Rfunction{resize} & Change width, anchoring on start, end or mid \\
    & \Rfunction{distance} & Separation between ranges (closest endpoints) \\
    & \Rfunction{restrict} & Clamp ranges to within some start and end \\
    & \Rfunction{flank} & Generate adjacent regions on start or end \\

    Set operations & \Rfunction{reduce} & 
        Merge overlapping and adjacent ranges \\
    & \Rfunction{intersect}, \Rfunction{union}, \Rfunction{setdiff} & 
        Set operations on reduced ranges \\
    & \Rfunction{pintersect}, \Rfunction{punion}, \Rfunction{psetdiff} & 
        Parallel set operations, on each \Robject{x[i]}, \Robject{y[i]} \\
    & \Rfunction{gaps}, \Rfunction{pgap} & 
        Find regions not covered by reduced ranges \\
    & \Rfunction{disjoin} & Ranges formed from union of endpoints \\

    Overlaps & \Rfunction{findOverlaps} & 
        Find all overlaps for each \Robject{x} in \Robject{y} \\
    & \Rfunction{countOverlaps} & 
        Count overlaps of each \Robject{x} range in \Robject{y} \\
    & \Rfunction{nearest} & 
        Find nearest neighbors (closest endpoints) \\
    & \Rfunction{precede}, \Rfunction{follow} & 
        Find nearest \Robject{y} that \Robject{x} precedes or follows \\
    & \Robject{x \%in\% y} & 
        Find ranges in \Robject{x} that overlap range in \Robject{y} \\

    Coverage & \Rfunction{coverage} & Count ranges covering each position \\

    Extraction & \Robject{r[i]} & Get or set by logical or numeric index \\
    & \Robject{r[[i]]} & 
        Get integer sequence from \Robject{start[i]} to \Robject{end[i]} \\
    & \Rfunction{subsetByOverlaps} & 
        Subset \Robject{x} for those that overlap in \Robject{y} \\
    & \Rfunction{head}, \Rfunction{tail}, \Rfunction{rev}, \Rfunction{rep} & 
        Conventional R semantics \\

    Split, combine & \Rfunction{split} & 
        Split ranges by a factor into a \Rclass{RangesList} \\
    & \Rfunction{c} & Concatenate two or more range objects \\
    \hline
  \end{tabular}
\end{table}


## Summarized Experiments

[SummarizedExperiment]()

 ![Alt SummarizedExperiment](fig/SummarizedExperiment.png)

- Comprehensive data structure that can be used to store expression and methylation data from microarrays or read counts from RNA-seq experiments, among others. 

- Can contain slots for one or more _omic_ datasets, feature annotation (e.g. genes, transcripts, SNPs, CpGs), individual phenotypes and experimental details, such as laboratory and experimental protocols. 

- As in an `ExpressionSet` a `SummarizedExperiment`, the rows of _omic_ data are features and columns are subjects. 

- Coordinate feature x sample 'assays' with row (feature) and column
  (sample) descriptions.

- 'assays' (similar to 'exprs' in `ExpressionSet`objects) can be any matrix-like object, including very large on-disk representations such as [HDF5Array]()

- 'assays' are annotated using `GenomicRanges`

- It is being deprecated


## Ranged Summarized Experiments

`SummarizedExperiment` is extended to `RangedSummarizedExperiment`, a child class that contains the annotation data of the features in a `GenomicRanges` object. An example dataset, stored as a `RangedSummarizedExperiment` is available in the `airway` package. This data represents an RNA sequencing experiment.

```{r load_airway}
library(airway)
data(airway)
airway
```

- Some aspects of the object are very similar to `ExpressionSet`, although with slightly different names and types. `colData` contains phenotype (sample) information, like `pData` for `ExpressionSet`. It returns a `DataFrame` instead of a data.frame:

```{r colData}
colData(airway)
``` 

- You can still use `$` to get a particular column:

```{r column}
airway$cell
``` 

- The measurement data are accessed by `assay` and `assays`. A `SummarizedExperiment` can contain multiple measurement matrices (all of the same dimension). You get all of them by assays and you select a particular one by `assay(OBJECT, NAME)` where you can see the names when you print the object or by using `assayNames`. In this case there is a single matrix called counts:

```{r counts}
assayNames(airway)

assays(airway)

head(assay(airway, "counts"))
```

- Annotation is a `GenomicRanges` object

```{r annot_sum_exp}
rowRanges(airway)
```

- Subset for only rows (e.g. features) which are in the interval 1 to 1Mb of chromosome 1

```{r subset_granges}
roi <- GRanges(seqnames="1", ranges=IRanges(start=1, end=1e6))
subsetByOverlaps(airway, roi)
```



## Multi Data Set

[MultiDataSet]()

   ![Alt MultiDataSet](fig/mds.png)

- Designed to encapsulate different types of datasets (including all classes in Bioconductor)
- It properly deals with non-complete cases situations
- Subsetting is easily performed in both: samples and features (using GenomicRanges)
- It allows to:
-- perform integration analysis with third party packages; 
-- create new methods and functions for omic data integration; 
-- encapsule new unimplemented data from any biological experiment

- [MultiAssayExperiment]() is another infrastructure (created by BioC developers) that can be used to manage multi-omic data


[Bioconductor]: https://bioconductor.org
[CRAN]: https://cran.r-project.org
[R]: https://cran.r-project.org
[DataSHIELD]: http://www.datashield.ac.uk/
[resourcer]: https://cran.r-project.org/web/packages/resourcer/index.html

[biocViews]: https://bioconductor.org/packages/
[HDF5Array]: https://bioconductor.org/packages/HDF5Array
[AnnotationDbi]: https://bioconductor.org/packages/AnnotationDbi
[AnnotationHub]: https://bioconductor.org/packages/AnnotationHub
[BSgenome.Hsapiens.UCSC.hg19]: https://bioconductor.org/packages/BSgenome.Hsapiens.UCSC.hg19
[BSgenome]: https://bioconductor.org/packages/BSgenome
[BiocParallel]: https://bioconductor.org/packages/BiocParallel
[Biostrings]: https://bioconductor.org/packages/Biostrings
[CNTools]: https://bioconductor.org/packages/CNTools
[ChIPQC]: https://bioconductor.org/packages/ChIPQC
[ChIPseeker]: https://bioconductor.org/packages/ChIPseeker
[DESeq2]: https://bioconductor.org/packages/DESeq2
[DiffBind]: https://bioconductor.org/packages/DiffBind
[GenomicAlignments]: https://bioconductor.org/packages/GenomicAlignments
[GenomicFeatures]: https://bioconductor.org/packages/GenomicFeatures
[GenomicFiles]: https://bioconductor.org/packages/GenomicFiles
[GenomicRanges]: https://bioconductor.org/packages/GenomicRanges
[Gviz]: https://bioconductor.org/packages/Gviz
[Homo.sapiens]: https://bioconductor.org/packages/Homo.sapiens
[IRanges]: https://bioconductor.org/packages/IRanges
[KEGGREST]: https://bioconductor.org/packages/KEGGREST
[OmicCircos]: https://bioconductor.org/packages/OmicCircos
[PSICQUIC]: https://bioconductor.org/packages/PSICQUIC
[Rsamtools]: https://bioconductor.org/packages/Rsamtools
[Rsubread]: https://bioconductor.org/packages/Rsubread
[ShortRead]: https://bioconductor.org/packages/ShortRead
[SomaticSignatures]: https://bioconductor.org/packages/SomaticSignatures
[SummarizedExperiment]: https://bioconductor.org/packages/SummarizedExperiment
[TxDb.Hsapiens.UCSC.hg19.knownGene]: https://bioconductor.org/packages/TxDb.Hsapiens.UCSC.hg19.knownGene
[VariantAnnotation]: https://bioconductor.org/packages/VariantAnnotation
[VariantFiltering]: https://bioconductor.org/packages/VariantFiltering
[VariantTools]: https://bioconductor.org/packages/VariantTools
[airway]: https://bioconductor.org/packages/airway
[biomaRt]: https://bioconductor.org/packages/biomaRt
[cn.mops]: https://bioconductor.org/packages/cn.mops
[csaw]: https://bioconductor.org/packages/csaw
[edgeR]: https://bioconductor.org/packages/edgeR
[ensemblVEP]: https://bioconductor.org/packages/ensemblVEP
[epivizr]: https://bioconductor.org/packages/epivizr
[ggbio]: https://bioconductor.org/packages/ggbio
[h5vc]: https://bioconductor.org/packages/h5vc
[limma]: https://bioconductor.org/packages/limma
[metagenomeSeq]: https://bioconductor.org/packages/metagenomeSeq
[org.Hs.eg.db]: https://bioconductor.org/packages/org.Hs.eg.db
[org.Sc.sgd.db]: https://bioconductor.org/packages/org.Sc.sgd.db
[phyloseq]: https://bioconductor.org/packages/phyloseq
[rtracklayer]: https://bioconductor.org/packages/rtracklayer
[snpStats]: https://bioconductor.org/packages/snpStats

[dplyr]: https://cran.r-project.org/package=dplyr
[data.table]: https://cran.r-project.org/package=data.table
[Rcpp]: https://cran.r-project.org/package=Rcpp
[kallisto]: https://pachterlab.github.io/kallisto

[MultiDataSet]: https://bioconductor.org/packages/MultiDataSet
[snpMatrix]: http://www.bioconductor.org/packages//2.7/bioc/html/snpMatrix.html
[PLINK]: http://pngu.mgh.harvard.edu/~purcell/plink/
[MultiAssayExperiment]: https://bioconductor.org/packages/MultiAssayExperiment
