<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>19 DSLite: DataSHIELD Implementation on Local Datasets | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD</title>
  <meta name="description" content="19 DSLite: DataSHIELD Implementation on Local Datasets | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="19 DSLite: DataSHIELD Implementation on Local Datasets | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="19 DSLite: DataSHIELD Implementation on Local Datasets | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD" />
  
  
  



<meta name="date" content="2021-01-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="exposome.html"/>
<link rel="next" href="package-info.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="citation.html"><a href="citation.html"><i class="fa fa-check"></i>Citation</a></li>
<li class="part"><span><b>I Preamble</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#what-you-will-learn"><i class="fa fa-check"></i><b>1.1</b> What you will learn</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#preliminaries"><i class="fa fa-check"></i><b>1.2</b> Preliminaries</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#acknowledgments"><i class="fa fa-check"></i><b>1.3</b> Acknowledgments</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="learnR.html"><a href="learnR.html"><i class="fa fa-check"></i><b>2</b> Learning R</a></li>
<li class="chapter" data-level="3" data-path="BioC.html"><a href="BioC.html"><i class="fa fa-check"></i><b>3</b> Bioconductor Data Structures</a>
<ul>
<li class="chapter" data-level="3.1" data-path="BioC.html"><a href="BioC.html#snp-array-data"><i class="fa fa-check"></i><b>3.1</b> SNP Array Data</a></li>
<li class="chapter" data-level="3.2" data-path="BioC.html"><a href="BioC.html#expression-sets"><i class="fa fa-check"></i><b>3.2</b> Expression Sets</a></li>
<li class="chapter" data-level="3.3" data-path="BioC.html"><a href="BioC.html#genomic-ranges"><i class="fa fa-check"></i><b>3.3</b> Genomic Ranges</a></li>
<li class="chapter" data-level="3.4" data-path="BioC.html"><a href="BioC.html#summarized-experiments"><i class="fa fa-check"></i><b>3.4</b> Summarized Experiments</a></li>
<li class="chapter" data-level="3.5" data-path="BioC.html"><a href="BioC.html#ranged-summarized-experiments"><i class="fa fa-check"></i><b>3.5</b> Ranged Summarized Experiments</a></li>
<li class="chapter" data-level="3.6" data-path="BioC.html"><a href="BioC.html#multi-data-set"><i class="fa fa-check"></i><b>3.6</b> Multi Data Set</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="opal.html"><a href="opal.html"><i class="fa fa-check"></i><b>4</b> Opal</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction.html"><a href="introduction.html#introduction"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="opal.html"><a href="opal.html#data-management"><i class="fa fa-check"></i><b>4.2</b> Data Management</a></li>
<li class="chapter" data-level="4.3" data-path="opal.html"><a href="opal.html#security"><i class="fa fa-check"></i><b>4.3</b> Security</a></li>
<li class="chapter" data-level="4.4" data-path="opal.html"><a href="opal.html#r-integration"><i class="fa fa-check"></i><b>4.4</b> R Integration</a></li>
<li class="chapter" data-level="4.5" data-path="opal.html"><a href="opal.html#opal-demo-site"><i class="fa fa-check"></i><b>4.5</b> Opal demo site</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="datashield.html"><a href="datashield.html"><i class="fa fa-check"></i><b>5</b> DataSHIELD</a>
<ul>
<li class="chapter" data-level="5.1" data-path="introduction.html"><a href="introduction.html#introduction"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="datashield.html"><a href="datashield.html#datashield-r-interface-dsi"><i class="fa fa-check"></i><b>5.2</b> DataSHIELD R Interface (DSI)</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="datashield.html"><a href="datashield.html#class-structures"><i class="fa fa-check"></i><b>5.2.1</b> Class Structures</a></li>
<li class="chapter" data-level="5.2.2" data-path="datashield.html"><a href="datashield.html#higher-level-functions"><i class="fa fa-check"></i><b>5.2.2</b> Higher Level Functions</a></li>
<li class="chapter" data-level="5.2.3" data-path="datashield.html"><a href="datashield.html#options"><i class="fa fa-check"></i><b>5.2.3</b> Options</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="datashield.html"><a href="datashield.html#datashieldopal-implementation"><i class="fa fa-check"></i><b>5.3</b> DataSHIELD/Opal Implementation</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="datashield.html"><a href="datashield.html#client"><i class="fa fa-check"></i><b>5.3.1</b> Client</a></li>
<li class="chapter" data-level="5.3.2" data-path="datashield.html"><a href="datashield.html#server"><i class="fa fa-check"></i><b>5.3.2</b> Server</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="datashield.html"><a href="datashield.html#demo"><i class="fa fa-check"></i><b>5.4</b> Demo</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="6" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>6</b> The Resources</a>
<ul>
<li class="chapter" data-level="6.1" data-path="resources.html"><a href="resources.html#concept"><i class="fa fa-check"></i><b>6.1</b> Concept</a></li>
<li class="chapter" data-level="6.2" data-path="resources.html"><a href="resources.html#types"><i class="fa fa-check"></i><b>6.2</b> Types</a></li>
<li class="chapter" data-level="6.3" data-path="resources.html"><a href="resources.html#definition"><i class="fa fa-check"></i><b>6.3</b> Definition</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="resourcer.html"><a href="resourcer.html"><i class="fa fa-check"></i><b>7</b> The resourcer Package</a>
<ul>
<li class="chapter" data-level="7.1" data-path="introduction.html"><a href="introduction.html#introduction"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="resourcer.html"><a href="resourcer.html#file-resources"><i class="fa fa-check"></i><b>7.2</b> File Resources</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="resourcer.html"><a href="resourcer.html#file-getter"><i class="fa fa-check"></i><b>7.2.1</b> File Getter</a></li>
<li class="chapter" data-level="7.2.2" data-path="resourcer.html"><a href="resourcer.html#file-data-format"><i class="fa fa-check"></i><b>7.2.2</b> File Data Format</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="resourcer.html"><a href="resourcer.html#database-resources"><i class="fa fa-check"></i><b>7.3</b> Database Resources</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="resourcer.html"><a href="resourcer.html#dbi-connectors"><i class="fa fa-check"></i><b>7.3.1</b> DBI Connectors</a></li>
<li class="chapter" data-level="7.3.2" data-path="resourcer.html"><a href="resourcer.html#use-dplyr"><i class="fa fa-check"></i><b>7.3.2</b> Use dplyr</a></li>
<li class="chapter" data-level="7.3.3" data-path="resourcer.html"><a href="resourcer.html#document-databases"><i class="fa fa-check"></i><b>7.3.3</b> Document Databases</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="resourcer.html"><a href="resourcer.html#computation-resources"><i class="fa fa-check"></i><b>7.4</b> Computation Resources</a></li>
<li class="chapter" data-level="7.5" data-path="resourcer.html"><a href="resourcer.html#interacting-with-r-resources"><i class="fa fa-check"></i><b>7.5</b> Interacting with R Resources</a></li>
<li class="chapter" data-level="7.6" data-path="resourcer.html"><a href="resourcer.html#extending_resources"><i class="fa fa-check"></i><b>7.6</b> Extending Resources</a></li>
<li class="chapter" data-level="7.7" data-path="resourcer.html"><a href="resourcer.html#resource-forms"><i class="fa fa-check"></i><b>7.7</b> Resource Forms</a></li>
<li class="chapter" data-level="7.8" data-path="resourcer.html"><a href="resourcer.html#examples-of-resources"><i class="fa fa-check"></i><b>7.8</b> Examples of Resources</a></li>
<li class="chapter" data-level="7.9" data-path="resourcer.html"><a href="resourcer.html#using-resources-with-datashield"><i class="fa fa-check"></i><b>7.9</b> Using Resources with DataSHIELD</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="resourcer.html"><a href="resourcer.html#tsv-file-into-a-tibble-or-data.frame"><i class="fa fa-check"></i><b>7.9.1</b> TSV File into a tibble or data.frame</a></li>
<li class="chapter" data-level="7.9.2" data-path="resourcer.html"><a href="resourcer.html#r-data-file-into-a-r-object"><i class="fa fa-check"></i><b>7.9.2</b> R Data File into a R object</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>III Resources Extensions</b></span></li>
<li class="chapter" data-level="8" data-path="omic-extension.html"><a href="omic-extension.html"><i class="fa fa-check"></i><b>8</b> Extension to VCF files to peform GWAS with Bioconductor</a></li>
<li class="chapter" data-level="9" data-path="shell-extension.html"><a href="shell-extension.html"><i class="fa fa-check"></i><b>9</b> Extension to secure shell programs: GWAS with PLINK</a></li>
<li class="chapter" data-level="10" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html"><i class="fa fa-check"></i><b>10</b> Extension to GA4GH and EGA (BAM/VCF files)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#upload-ga4gh-and-ega-resources-into-opal"><i class="fa fa-check"></i><b>10.1</b> Upload GA4GH and EGA resources into OPAL</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#manually"><i class="fa fa-check"></i><b>10.1.1</b> Manually</a></li>
<li class="chapter" data-level="10.1.2" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#using-r"><i class="fa fa-check"></i><b>10.1.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#adding-a-genomic-resource"><i class="fa fa-check"></i><b>10.2</b> Adding a genomic resource</a></li>
<li class="chapter" data-level="10.3" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#descriptive-omic-data-analysis"><i class="fa fa-check"></i><b>10.3</b> Descriptive omic data analysis</a></li>
</ul></li>
<li class="part"><span><b>IV Workflows</b></span></li>
<li class="chapter" data-level="11" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>11</b> Setup</a>
<ul>
<li class="chapter" data-level="11.1" data-path="setup.html"><a href="setup.html#providing-datashield-packages-in-the-opal-server"><i class="fa fa-check"></i><b>11.1</b> Providing DataSHIELD packages in the opal server</a></li>
<li class="chapter" data-level="11.2" data-path="setup.html"><a href="setup.html#required-r-packages-in-the-client-site-e.g.-local-machine"><i class="fa fa-check"></i><b>11.2</b> Required R Packages in the client site (e.g. local machine)</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="basic-statistical-analyses.html"><a href="basic-statistical-analyses.html"><i class="fa fa-check"></i><b>12</b> Basic statistical analyses</a>
<ul>
<li class="chapter" data-level="12.1" data-path="basic-statistical-analyses.html"><a href="basic-statistical-analyses.html#analysis-from-a-single-study"><i class="fa fa-check"></i><b>12.1</b> Analysis from a single study</a></li>
<li class="chapter" data-level="12.2" data-path="basic-statistical-analyses.html"><a href="basic-statistical-analyses.html#analysis-from-a-multiple-studies"><i class="fa fa-check"></i><b>12.2</b> Analysis from a multiple studies</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Omic.html"><a href="Omic.html"><i class="fa fa-check"></i><b>13</b> Omic data analysis: types of implemented analyses</a></li>
<li class="chapter" data-level="14" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html"><i class="fa fa-check"></i><b>14</b> Differential gene expression (DGE) analysis</a></li>
<li class="chapter" data-level="15" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html"><i class="fa fa-check"></i><b>15</b> Epigenome-wide association analysis (EWAS)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#single-cpg-analysis"><i class="fa fa-check"></i><b>15.1</b> Single CpG analysis</a></li>
<li class="chapter" data-level="15.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#multiple-cpg-analysis"><i class="fa fa-check"></i><b>15.2</b> Multiple CpG analysis</a></li>
<li class="chapter" data-level="15.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#adjusting-for-surrogate-variables"><i class="fa fa-check"></i><b>15.3</b> Adjusting for Surrogate Variables</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="genome-wide-association-analyses-gwas.html"><a href="genome-wide-association-analyses-gwas.html"><i class="fa fa-check"></i><b>16</b> Genome-wide association analyses (GWAS)</a>
<ul>
<li class="chapter" data-level="16.1" data-path="genome-wide-association-analyses-gwas.html"><a href="genome-wide-association-analyses-gwas.html#gwasBioc"><i class="fa fa-check"></i><b>16.1</b> GWAS with Bioconductor</a></li>
<li class="chapter" data-level="16.2" data-path="genome-wide-association-analyses-gwas.html"><a href="genome-wide-association-analyses-gwas.html#gwas-with-plink"><i class="fa fa-check"></i><b>16.2</b> GWAS with PLINK</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="GIS.html"><a href="GIS.html"><i class="fa fa-check"></i><b>17</b> Geospatial data analysis</a>
<ul>
<li class="chapter" data-level="17.1" data-path="GIS.html"><a href="GIS.html#introducing-the-analysis"><i class="fa fa-check"></i><b>17.1</b> Introducing the analysis</a></li>
<li class="chapter" data-level="17.2" data-path="GIS.html"><a href="GIS.html#setting-up-the-analysis"><i class="fa fa-check"></i><b>17.2</b> Setting up the analysis</a></li>
<li class="chapter" data-level="17.3" data-path="GIS.html"><a href="GIS.html#data-manipulation"><i class="fa fa-check"></i><b>17.3</b> Data manipulation</a></li>
<li class="chapter" data-level="17.4" data-path="GIS.html"><a href="GIS.html#generating-the-final-result"><i class="fa fa-check"></i><b>17.4</b> Generating the final result</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="exposome.html"><a href="exposome.html"><i class="fa fa-check"></i><b>18</b> Exposome data analysis</a>
<ul>
<li class="chapter" data-level="18.1" data-path="exposome.html"><a href="exposome.html#loading-the-exposome-set"><i class="fa fa-check"></i><b>18.1</b> Loading the Exposome Set</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="exposome.html"><a href="exposome.html#from-tables-to-exposomeset"><i class="fa fa-check"></i><b>18.1.1</b> From tables to <code>ExposomeSet</code></a></li>
<li class="chapter" data-level="18.1.2" data-path="exposome.html"><a href="exposome.html#from-resource-to-exposomeset"><i class="fa fa-check"></i><b>18.1.2</b> From resource to <code>ExposomeSet</code></a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="exposome.html"><a href="exposome.html#exploring-the-loaded-dataset"><i class="fa fa-check"></i><b>18.2</b> Exploring the loaded dataset</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="exposome.html"><a href="exposome.html#family-names"><i class="fa fa-check"></i><b>18.2.1</b> Family names</a></li>
<li class="chapter" data-level="18.2.2" data-path="exposome.html"><a href="exposome.html#exposures-and-phenotypes-names"><i class="fa fa-check"></i><b>18.2.2</b> Exposures and phenotypes names</a></li>
<li class="chapter" data-level="18.2.3" data-path="exposome.html"><a href="exposome.html#summary-of-variables"><i class="fa fa-check"></i><b>18.2.3</b> Summary of variables</a></li>
<li class="chapter" data-level="18.2.4" data-path="exposome.html"><a href="exposome.html#missing-data"><i class="fa fa-check"></i><b>18.2.4</b> Missing data</a></li>
<li class="chapter" data-level="18.2.5" data-path="exposome.html"><a href="exposome.html#exposures-normality"><i class="fa fa-check"></i><b>18.2.5</b> Exposures Normality</a></li>
<li class="chapter" data-level="18.2.6" data-path="exposome.html"><a href="exposome.html#exposures-behaviour"><i class="fa fa-check"></i><b>18.2.6</b> Exposures Behaviour</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="exposome.html"><a href="exposome.html#exposures-imputation"><i class="fa fa-check"></i><b>18.3</b> Exposures Imputation</a></li>
<li class="chapter" data-level="18.4" data-path="exposome.html"><a href="exposome.html#performing-an-exwas"><i class="fa fa-check"></i><b>18.4</b> Performing an ExWAS</a></li>
<li class="chapter" data-level="18.5" data-path="exposome.html"><a href="exposome.html#exposures-pca"><i class="fa fa-check"></i><b>18.5</b> Exposures PCA</a></li>
<li class="chapter" data-level="18.6" data-path="exposome.html"><a href="exposome.html#exposures-correlation"><i class="fa fa-check"></i><b>18.6</b> Exposures Correlation</a></li>
</ul></li>
<li class="part"><span><b>V Developers</b></span></li>
<li class="chapter" data-level="19" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html"><i class="fa fa-check"></i><b>19</b> DSLite: DataSHIELD Implementation on Local Datasets</a>
<ul>
<li class="chapter" data-level="19.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#development-environment-setup"><i class="fa fa-check"></i><b>19.1</b> Development Environment Setup</a>
<ul>
<li class="chapter" data-level="19.1.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-packages"><i class="fa fa-check"></i><b>19.1.1</b> DataSHIELD Packages</a></li>
<li class="chapter" data-level="19.1.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#test-datasets"><i class="fa fa-check"></i><b>19.1.2</b> Test Datasets</a></li>
<li class="chapter" data-level="19.1.3" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#r-package-development-tools"><i class="fa fa-check"></i><b>19.1.3</b> R Package Development Tools</a></li>
</ul></li>
<li class="chapter" data-level="19.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-development-flow"><i class="fa fa-check"></i><b>19.2</b> DataSHIELD Development Flow</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#dsliteserver"><i class="fa fa-check"></i><b>19.2.1</b> DSLiteServer</a></li>
<li class="chapter" data-level="19.2.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-configuration"><i class="fa fa-check"></i><b>19.2.2</b> DataSHIELD Configuration</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-sessions"><i class="fa fa-check"></i><b>19.3</b> DataSHIELD Sessions</a></li>
<li class="chapter" data-level="19.4" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#debugging"><i class="fa fa-check"></i><b>19.4</b> Debugging</a></li>
<li class="chapter" data-level="19.5" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#limitations"><i class="fa fa-check"></i><b>19.5</b> Limitations</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#function-parameters-parser"><i class="fa fa-check"></i><b>19.5.1</b> Function Parameters Parser</a></li>
<li class="chapter" data-level="19.5.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#server-side-environments"><i class="fa fa-check"></i><b>19.5.2</b> Server Side Environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="package-info.html"><a href="package-info.html"><i class="fa fa-check"></i><b>20</b> Creating DataSHIELD packages</a>
<ul>
<li class="chapter" data-level="20.1" data-path="package-info.html"><a href="package-info.html#example-of-datashield-package-development"><i class="fa fa-check"></i><b>20.1</b> Example of DataSHIELD package development</a>
<ul>
<li class="chapter" data-level="20.1.1" data-path="package-info.html"><a href="package-info.html#fandfiles"><i class="fa fa-check"></i><b>20.1.1</b> Folders and files</a></li>
<li class="chapter" data-level="20.1.2" data-path="package-info.html"><a href="package-info.html#development-of-a-function"><i class="fa fa-check"></i><b>20.1.2</b> Development of a function</a></li>
<li class="chapter" data-level="20.1.3" data-path="package-info.html"><a href="package-info.html#testing-of-the-function"><i class="fa fa-check"></i><b>20.1.3</b> Testing of the function</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html"><i class="fa fa-check"></i><b>21</b> Tips and tricks</a>
<ul>
<li class="chapter" data-level="21.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-create-a-new-project-into-opal"><i class="fa fa-check"></i><b>21.1</b> How to create a new project into OPAL</a>
<ul>
<li class="chapter" data-level="21.1.1" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#manually"><i class="fa fa-check"></i><b>21.1.1</b> Manually</a></li>
<li class="chapter" data-level="21.1.2" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#using-r"><i class="fa fa-check"></i><b>21.1.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-upload-a-new-resource-into-opal"><i class="fa fa-check"></i><b>21.2</b> How to upload a new resource into OPAL</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#create_manually"><i class="fa fa-check"></i><b>21.2.1</b> Manually</a></li>
<li class="chapter" data-level="21.2.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#using-r-1"><i class="fa fa-check"></i><b>21.2.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-install-datashield-packages-into-opal-server"><i class="fa fa-check"></i><b>21.3</b> How to install DataSHIELD packages into OPAL server</a>
<ul>
<li class="chapter" data-level="21.3.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#manually-1"><i class="fa fa-check"></i><b>21.3.1</b> Manually</a></li>
<li class="chapter" data-level="21.3.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#using-r-2"><i class="fa fa-check"></i><b>21.3.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-install-r-packages-into-opal-server"><i class="fa fa-check"></i><b>21.4</b> How to install R packages into OPAL server</a>
<ul>
<li class="chapter" data-level="21.4.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#manually-2"><i class="fa fa-check"></i><b>21.4.1</b> Manually</a></li>
<li class="chapter" data-level="21.4.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#using-r-3"><i class="fa fa-check"></i><b>21.4.2</b> Using R</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="advancedtechnical-questions.html"><a href="advancedtechnical-questions.html"><i class="fa fa-check"></i><b>22</b> Advanced/technical questions</a></li>
<li class="part"><span><b>VI Appendix</b></span></li>
<li class="chapter" data-level="23" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>23</b> Session Info</a></li>
<li class="chapter" data-level="24" data-path="contributors.html"><a href="contributors.html"><i class="fa fa-check"></i><b>24</b> Contributors</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dslite-datashield-implementation-on-local-datasets" class="section level1" number="19">
<h1><span class="header-section-number">19</span> DSLite: DataSHIELD Implementation on Local Datasets</h1>
<p>DSLite is a serverless <a href="https://github.com/datashield/DSI/">DataSHIELD Interface (DSI)</a> implementation whose purpose is to mimic
the behavior of a distant (virtualized or barebone) data repository server (see <a href="https://github.com/datashield/DSOpal">DSOpal</a> for instance). The datasets that are being analyzed must be fully accessible in the local environment and then the non-disclosive constraint of the analysis is not relevant for DSLite: some DSLite functionalities allow inspection of what is under the hood of the DataSHIELD computation nodes, making it a perfect tool for DataSHIELD analysis package developers.</p>
<div id="development-environment-setup" class="section level2" number="19.1">
<h2><span class="header-section-number">19.1</span> Development Environment Setup</h2>
<div id="datashield-packages" class="section level3" number="19.1.1">
<h3><span class="header-section-number">19.1.1</span> DataSHIELD Packages</h3>
<p>Both client and server side packages must be installed in your local R session. The entry point is still the client side package and DSLite will automatically load the corresponding server side package on DataSHIELD aggregate and assignment functions call, based on the DataSHIELD configuration. The minimum required packages are:</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="dslite-datashield-implementation-on-local-datasets.html#cb336-1"></a><span class="kw">install.packages</span>(<span class="kw">c</span>(<span class="st">&quot;resourcer&quot;</span>, <span class="st">&quot;DSLite&quot;</span>), <span class="dt">dependencies =</span> <span class="ot">TRUE</span>)</span>
<span id="cb336-2"><a href="dslite-datashield-implementation-on-local-datasets.html#cb336-2"></a><span class="kw">install.packages</span>(<span class="st">&quot;dsBase&quot;</span>, <span class="dt">repos =</span> <span class="kw">c</span>(<span class="st">&quot;https://cloud.r-project.org&quot;</span>, <span class="st">&quot;https://cran.obiba.org&quot;</span>), <span class="dt">dependencies =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
</div>
<div id="test-datasets" class="section level3" number="19.1.2">
<h3><span class="header-section-number">19.1.2</span> Test Datasets</h3>
<p>DSLite comes with a set of datasets that can be easily loaded. You can also provide your own to illustrate a specific data analysis function.</p>
</div>
<div id="r-package-development-tools" class="section level3" number="19.1.3">
<h3><span class="header-section-number">19.1.3</span> R Package Development Tools</h3>
<p>We recommend using the following tools to facilitate R package development:</p>
<ul>
<li><a href="https://www.rdocumentation.org/packages/devtools">devtools</a>, the collection of package development tools,</li>
<li><a href="https://www.rdocumentation.org/packages/usethis">usethis</a>, automate package and project setup tasks that are otherwise performed manually,</li>
<li><a href="https://www.rdocumentation.org/packages/testthat">testthat</a>, for unit testing,</li>
<li><a href="https://www.rdocumentation.org/packages/roxygen2">roxygen2</a>, for writing documentation in-line with code,</li>
<li><a href="https://rstudio.com/">Rstudio</a>, the R editor that integrates the tools mentioned above and more.</li>
</ul>
</div>
</div>
<div id="datashield-development-flow" class="section level2" number="19.2">
<h2><span class="header-section-number">19.2</span> DataSHIELD Development Flow</h2>
<p>The typical development flow, using DSLite, is:</p>
<ol style="list-style-type: decimal">
<li>Build and install your client and/or server side DataSHIELD packages.</li>
<li>Create a new DSLiteServer object instance, refering test datasets. Use or alter the default DataSHIELD configuration.</li>
<li>Test your DataSHIELD client/server functions.</li>
<li>Debug DataSHIELD server nodes using DSLiteServer methods.</li>
</ol>
<div id="dsliteserver" class="section level3" number="19.2.1">
<h3><span class="header-section-number">19.2.1</span> DSLiteServer</h3>
<p>After your client and/or server side DataSHIELD packages have been built and installed, a new DSLiteServer object instance must be created.</p>
<p>Some DSLiteServer methods can be used to verify or modify the DSLiteServer behaviour:</p>
<ul>
<li><code>DSLiteServer$strict()</code></li>
<li><code>DSLiteServer$home()</code></li>
</ul>
<p>See the R documentation of the DSLiteServer class for details.</p>
<p>As an example:</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="dslite-datashield-implementation-on-local-datasets.html#cb337-1"></a><span class="kw">library</span>(DSLite)</span>
<span id="cb337-2"><a href="dslite-datashield-implementation-on-local-datasets.html#cb337-2"></a><span class="co"># prepare test data in a light DS server</span></span>
<span id="cb337-3"><a href="dslite-datashield-implementation-on-local-datasets.html#cb337-3"></a><span class="kw">data</span>(<span class="st">&quot;CNSIM1&quot;</span>)</span>
<span id="cb337-4"><a href="dslite-datashield-implementation-on-local-datasets.html#cb337-4"></a><span class="kw">data</span>(<span class="st">&quot;CNSIM2&quot;</span>)</span>
<span id="cb337-5"><a href="dslite-datashield-implementation-on-local-datasets.html#cb337-5"></a><span class="kw">data</span>(<span class="st">&quot;CNSIM3&quot;</span>)</span>
<span id="cb337-6"><a href="dslite-datashield-implementation-on-local-datasets.html#cb337-6"></a>dslite.server &lt;-<span class="st"> </span><span class="kw">newDSLiteServer</span>(<span class="dt">tables=</span><span class="kw">list</span>(<span class="dt">CNSIM1=</span>CNSIM1, <span class="dt">CNSIM2=</span>CNSIM2, <span class="dt">CNSIM3=</span>CNSIM3))</span>
<span id="cb337-7"><a href="dslite-datashield-implementation-on-local-datasets.html#cb337-7"></a><span class="co"># load corresponding DataSHIELD login data</span></span>
<span id="cb337-8"><a href="dslite-datashield-implementation-on-local-datasets.html#cb337-8"></a><span class="kw">data</span>(<span class="st">&quot;logindata.dslite.cnsim&quot;</span>)</span></code></pre></div>
<p>The previous example can be simplified using the set-up functions based on the provided test datasets:</p>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="dslite-datashield-implementation-on-local-datasets.html#cb338-1"></a><span class="kw">library</span>(DSLite)</span>
<span id="cb338-2"><a href="dslite-datashield-implementation-on-local-datasets.html#cb338-2"></a><span class="co"># load CNSIM test data</span></span>
<span id="cb338-3"><a href="dslite-datashield-implementation-on-local-datasets.html#cb338-3"></a>logindata.dslite.cnsim &lt;-<span class="st"> </span><span class="kw">setupCNSIMTest</span>()</span></code></pre></div>
</div>
<div id="datashield-configuration" class="section level3" number="19.2.2">
<h3><span class="header-section-number">19.2.2</span> DataSHIELD Configuration</h3>
<p>The DataSHIELD configuration (aggregate and assign functions, R options) is automatically discovered by inspecting the R packages installed and having some DataSHIELD settings defined, either in their <code>DESCRIPTION</code> file or in a <code>DATASHIELD</code> file.</p>
<p>This default configuration extracting function is:</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="dslite-datashield-implementation-on-local-datasets.html#cb339-1"></a>DSLite<span class="op">::</span><span class="kw">defaultDSConfiguration</span>()</span></code></pre></div>
<p>The list of the DataSHIELD R packages to be inspected (or excluded) when building the default configuration can be specified as parameters of <code>defaultDSConfiguration()</code>.</p>
<p>The DataSHIELD configuration can be specified at DSLiteServer creation time or afterwards with some DSLiteServer methods that can be used to verify or modify the DSLiteServer configuration:</p>
<ul>
<li><code>DSLiteServer$config()</code></li>
<li><code>DSLiteServer$aggregateMethods()</code></li>
<li><code>DSLiteServer$aggregateMethod()</code></li>
<li><code>DSLiteServer$assignMethods()</code></li>
<li><code>DSLiteServer$assignMethod()</code></li>
<li><code>DSLiteServer$options()</code></li>
<li><code>DSLiteServer$option()</code></li>
</ul>
<p>See the R documentation of the DSLiteServer class for details.</p>
<p>As an example:</p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="dslite-datashield-implementation-on-local-datasets.html#cb340-1"></a><span class="co"># verify configuration</span></span>
<span id="cb340-2"><a href="dslite-datashield-implementation-on-local-datasets.html#cb340-2"></a>dslite.server<span class="op">$</span><span class="kw">config</span>()</span></code></pre></div>
</div>
</div>
<div id="datashield-sessions" class="section level2" number="19.3">
<h2><span class="header-section-number">19.3</span> DataSHIELD Sessions</h2>
<p>The following figure illustrates a setup where a single DSLiteServer holds several data frames and is used by two different DataSHIELD Connection (<a href="https://github.com/datashield/DSI">DSConnection</a>) objects. All these objects live in the same R environment (usually the Global Environment). The “server” is responsible for managing DataSHIELD sessions that are implemented as distinct R environments inside of which R symbols are assigned and R functions are evaluated. Using the <a href="https://adv-r.hadley.nz/environments.html">R environment</a> paradigm ensures that the different DataSHIELD execution context (client and servers) are contained and exclusive from each other.</p>
<div class="figure">
<img src="https://raw.githubusercontent.com/datashield/DSLite/master/inst/images/dslite.png" alt="" />
<p class="caption">DSLite architecture</p>
</div>
<p>After performing the login DataSHIELD phase, the DSLiteServer holds the different DataSHIELD server side sessions, i.e. R environments identified by an ID. These IDs are also stored within the DataSHIELD connection objects that are the result of the <code>datashield.login()</code> call. The following example shows how to access these session IDs:</p>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="dslite-datashield-implementation-on-local-datasets.html#cb341-1"></a><span class="co"># datashield logins and assignments</span></span>
<span id="cb341-2"><a href="dslite-datashield-implementation-on-local-datasets.html#cb341-2"></a>conns &lt;-<span class="st"> </span><span class="kw">datashield.login</span>(logindata.dslite.cnsim, <span class="dt">assign=</span><span class="ot">TRUE</span>)</span>
<span id="cb341-3"><a href="dslite-datashield-implementation-on-local-datasets.html#cb341-3"></a><span class="co"># get the session ID of &quot;sim1&quot; node connection object</span></span>
<span id="cb341-4"><a href="dslite-datashield-implementation-on-local-datasets.html#cb341-4"></a>conns<span class="op">$</span>sim1<span class="op">@</span>sid</span>
<span id="cb341-5"><a href="dslite-datashield-implementation-on-local-datasets.html#cb341-5"></a><span class="co"># the same ID is in the DSLiteServer</span></span>
<span id="cb341-6"><a href="dslite-datashield-implementation-on-local-datasets.html#cb341-6"></a>dslite.server<span class="op">$</span><span class="kw">hasSession</span>(conns<span class="op">$</span>sim1<span class="op">@</span>sid)</span></code></pre></div>
</div>
<div id="debugging" class="section level2" number="19.4">
<h2><span class="header-section-number">19.4</span> Debugging</h2>
<p>Thanks to the DSLiteServer capability of allowing its configuration to be modified at any time, it is possible to add some debugging functions without polluting the DataSHIELD package you are developing.</p>
<p>For instance, this code adds an aggregate function <code>print()</code>:</p>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="dslite-datashield-implementation-on-local-datasets.html#cb342-1"></a><span class="co"># add a print method to configuration</span></span>
<span id="cb342-2"><a href="dslite-datashield-implementation-on-local-datasets.html#cb342-2"></a>dslite.server<span class="op">$</span><span class="kw">aggregateMethod</span>(<span class="st">&quot;print&quot;</span>, <span class="cf">function</span>(x){ <span class="kw">print</span>(x) })</span>
<span id="cb342-3"><a href="dslite-datashield-implementation-on-local-datasets.html#cb342-3"></a><span class="co"># and use it to print the D symbol</span></span>
<span id="cb342-4"><a href="dslite-datashield-implementation-on-local-datasets.html#cb342-4"></a><span class="kw">datashield.aggregate</span>(conns, <span class="kw">quote</span>(<span class="kw">print</span>(D)))</span></code></pre></div>
<p>Another option is to get a symbol value from the server into the client environment. This can be very helpful for complex data structures. The following example illustrates usage of a shortcut function that iterates over all the connection objects and get the corresponding symbol value:</p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="dslite-datashield-implementation-on-local-datasets.html#cb343-1"></a><span class="co"># get data represented by symbol D for each DataSHIELD connection</span></span>
<span id="cb343-2"><a href="dslite-datashield-implementation-on-local-datasets.html#cb343-2"></a>data &lt;-<span class="st"> </span><span class="kw">getDSLiteData</span>(conns, <span class="st">&quot;D&quot;</span>)</span>
<span id="cb343-3"><a href="dslite-datashield-implementation-on-local-datasets.html#cb343-3"></a><span class="co"># get data represented by symbol D from a specific DataSHIELD connection</span></span>
<span id="cb343-4"><a href="dslite-datashield-implementation-on-local-datasets.html#cb343-4"></a>data1 &lt;-<span class="st"> </span><span class="kw">getDSLiteData</span>(conns<span class="op">$</span>sim1, <span class="st">&quot;D&quot;</span>)</span></code></pre></div>
</div>
<div id="limitations" class="section level2" number="19.5">
<h2><span class="header-section-number">19.5</span> Limitations</h2>
<div id="function-parameters-parser" class="section level3" number="19.5.1">
<h3><span class="header-section-number">19.5.1</span> Function Parameters Parser</h3>
<p>The main difference with a regular DSI implementation (such as the one of DSOpal) is that the arguments of the DataSHIELD functional calls are not parsed in DSLite. The only R language element that is inspected and handled is the name of the functions, that are replaced by the ones defined in the DataSHIELD configuration.</p>
<p>For instance the following expression, which includes a function call in the formula, is valid for the DSLiteServer but not for Opal:</p>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="dslite-datashield-implementation-on-local-datasets.html#cb344-1"></a><span class="kw">someregression</span>(D<span class="op">$</span>height <span class="op">~</span><span class="st"> </span>D<span class="op">$</span>diameter <span class="op">+</span><span class="st"> </span><span class="kw">poly</span>(D<span class="op">$</span>length,<span class="dv">3</span>,<span class="dt">raw=</span><span class="ot">TRUE</span>))</span></code></pre></div>
<p>As a consequence, DataSHIELD R package development can take advantage of DSLite flexibility for rapid development but will never replace testing on regular DataSHIELD infrastructure using DSOpal.</p>
</div>
<div id="server-side-environments" class="section level3" number="19.5.2">
<h3><span class="header-section-number">19.5.2</span> Server Side Environments</h3>
<p>For each of the DataSHIELD nodes, the server side code is evaluated within an environment that has no parent, i.e. detached from the global environment where the client code is executed. Some R functions have a parameter that allows to specify to which environment they apply, for instance <code>assign()</code>, <code>get()</code>, <code>eval()</code>, <code>as.formula()</code>, etc. Their <code>env</code> (or <code>envir</code>) parameter default value is <code>parent.frame()</code> which is the global environment when executed in Opal’s R server, because it is the parent frame of the package’s namespace where the function is defined. In DSLiteServer, the parent frame must be the environment where the server code is evaluated. In order to be consistent between these two execution contexts (Opal R server and DSLiteServer), you must specify the <code>env</code> (or <code>envir</code>) value explicitly to be <code>parent.frame()</code>, which is the parent frame of the block being executed (either the global environment in Opal context, or the environment defined in DSLiteServer).</p>
<p>Example of a valid server side piece of code that assigns a value to a symbol in the DataSHIELD server’s environment (being the Opal R server’s global environment or a DSLiteServer’s environment):</p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="dslite-datashield-implementation-on-local-datasets.html#cb345-1"></a>base<span class="op">::</span><span class="kw">assign</span>(<span class="dt">x =</span> <span class="st">&quot;D&quot;</span>, <span class="dt">value =</span> someValue, <span class="dt">envir =</span> <span class="kw">parent.frame</span>())</span></code></pre></div>
<p>See also the <a href="http://adv-r.had.co.nz/Environments.html">Advanced R - Environments</a> documentation to learn more about environments.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="exposome.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="package-info.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
