<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>18 Exposome data analysis | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD</title>
  <meta name="description" content="18 Exposome data analysis | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="18 Exposome data analysis | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="18 Exposome data analysis | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD" />
  
  
  



<meta name="date" content="2021-01-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="GIS.html"/>
<link rel="next" href="dslite-datashield-implementation-on-local-datasets.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="citation.html"><a href="citation.html"><i class="fa fa-check"></i>Citation</a></li>
<li class="part"><span><b>I Preamble</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#what-you-will-learn"><i class="fa fa-check"></i><b>1.1</b> What you will learn</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#preliminaries"><i class="fa fa-check"></i><b>1.2</b> Preliminaries</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#acknowledgments"><i class="fa fa-check"></i><b>1.3</b> Acknowledgments</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="learnR.html"><a href="learnR.html"><i class="fa fa-check"></i><b>2</b> Learning R</a></li>
<li class="chapter" data-level="3" data-path="BioC.html"><a href="BioC.html"><i class="fa fa-check"></i><b>3</b> Bioconductor Data Structures</a>
<ul>
<li class="chapter" data-level="3.1" data-path="BioC.html"><a href="BioC.html#snp-array-data"><i class="fa fa-check"></i><b>3.1</b> SNP Array Data</a></li>
<li class="chapter" data-level="3.2" data-path="BioC.html"><a href="BioC.html#expression-sets"><i class="fa fa-check"></i><b>3.2</b> Expression Sets</a></li>
<li class="chapter" data-level="3.3" data-path="BioC.html"><a href="BioC.html#genomic-ranges"><i class="fa fa-check"></i><b>3.3</b> Genomic Ranges</a></li>
<li class="chapter" data-level="3.4" data-path="BioC.html"><a href="BioC.html#summarized-experiments"><i class="fa fa-check"></i><b>3.4</b> Summarized Experiments</a></li>
<li class="chapter" data-level="3.5" data-path="BioC.html"><a href="BioC.html#ranged-summarized-experiments"><i class="fa fa-check"></i><b>3.5</b> Ranged Summarized Experiments</a></li>
<li class="chapter" data-level="3.6" data-path="BioC.html"><a href="BioC.html#multi-data-set"><i class="fa fa-check"></i><b>3.6</b> Multi Data Set</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="opal.html"><a href="opal.html"><i class="fa fa-check"></i><b>4</b> Opal</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction.html"><a href="introduction.html#introduction"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="opal.html"><a href="opal.html#data-management"><i class="fa fa-check"></i><b>4.2</b> Data Management</a></li>
<li class="chapter" data-level="4.3" data-path="opal.html"><a href="opal.html#security"><i class="fa fa-check"></i><b>4.3</b> Security</a></li>
<li class="chapter" data-level="4.4" data-path="opal.html"><a href="opal.html#r-integration"><i class="fa fa-check"></i><b>4.4</b> R Integration</a></li>
<li class="chapter" data-level="4.5" data-path="opal.html"><a href="opal.html#opal-demo-site"><i class="fa fa-check"></i><b>4.5</b> Opal demo site</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="datashield.html"><a href="datashield.html"><i class="fa fa-check"></i><b>5</b> DataSHIELD</a>
<ul>
<li class="chapter" data-level="5.1" data-path="introduction.html"><a href="introduction.html#introduction"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="datashield.html"><a href="datashield.html#datashield-r-interface-dsi"><i class="fa fa-check"></i><b>5.2</b> DataSHIELD R Interface (DSI)</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="datashield.html"><a href="datashield.html#class-structures"><i class="fa fa-check"></i><b>5.2.1</b> Class Structures</a></li>
<li class="chapter" data-level="5.2.2" data-path="datashield.html"><a href="datashield.html#higher-level-functions"><i class="fa fa-check"></i><b>5.2.2</b> Higher Level Functions</a></li>
<li class="chapter" data-level="5.2.3" data-path="datashield.html"><a href="datashield.html#options"><i class="fa fa-check"></i><b>5.2.3</b> Options</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="datashield.html"><a href="datashield.html#datashieldopal-implementation"><i class="fa fa-check"></i><b>5.3</b> DataSHIELD/Opal Implementation</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="datashield.html"><a href="datashield.html#client"><i class="fa fa-check"></i><b>5.3.1</b> Client</a></li>
<li class="chapter" data-level="5.3.2" data-path="datashield.html"><a href="datashield.html#server"><i class="fa fa-check"></i><b>5.3.2</b> Server</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="datashield.html"><a href="datashield.html#demo"><i class="fa fa-check"></i><b>5.4</b> Demo</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="6" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>6</b> The Resources</a>
<ul>
<li class="chapter" data-level="6.1" data-path="resources.html"><a href="resources.html#concept"><i class="fa fa-check"></i><b>6.1</b> Concept</a></li>
<li class="chapter" data-level="6.2" data-path="resources.html"><a href="resources.html#types"><i class="fa fa-check"></i><b>6.2</b> Types</a></li>
<li class="chapter" data-level="6.3" data-path="resources.html"><a href="resources.html#definition"><i class="fa fa-check"></i><b>6.3</b> Definition</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="resourcer.html"><a href="resourcer.html"><i class="fa fa-check"></i><b>7</b> The resourcer Package</a>
<ul>
<li class="chapter" data-level="7.1" data-path="introduction.html"><a href="introduction.html#introduction"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="resourcer.html"><a href="resourcer.html#file-resources"><i class="fa fa-check"></i><b>7.2</b> File Resources</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="resourcer.html"><a href="resourcer.html#file-getter"><i class="fa fa-check"></i><b>7.2.1</b> File Getter</a></li>
<li class="chapter" data-level="7.2.2" data-path="resourcer.html"><a href="resourcer.html#file-data-format"><i class="fa fa-check"></i><b>7.2.2</b> File Data Format</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="resourcer.html"><a href="resourcer.html#database-resources"><i class="fa fa-check"></i><b>7.3</b> Database Resources</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="resourcer.html"><a href="resourcer.html#dbi-connectors"><i class="fa fa-check"></i><b>7.3.1</b> DBI Connectors</a></li>
<li class="chapter" data-level="7.3.2" data-path="resourcer.html"><a href="resourcer.html#use-dplyr"><i class="fa fa-check"></i><b>7.3.2</b> Use dplyr</a></li>
<li class="chapter" data-level="7.3.3" data-path="resourcer.html"><a href="resourcer.html#document-databases"><i class="fa fa-check"></i><b>7.3.3</b> Document Databases</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="resourcer.html"><a href="resourcer.html#computation-resources"><i class="fa fa-check"></i><b>7.4</b> Computation Resources</a></li>
<li class="chapter" data-level="7.5" data-path="resourcer.html"><a href="resourcer.html#interacting-with-r-resources"><i class="fa fa-check"></i><b>7.5</b> Interacting with R Resources</a></li>
<li class="chapter" data-level="7.6" data-path="resourcer.html"><a href="resourcer.html#extending_resources"><i class="fa fa-check"></i><b>7.6</b> Extending Resources</a></li>
<li class="chapter" data-level="7.7" data-path="resourcer.html"><a href="resourcer.html#resource-forms"><i class="fa fa-check"></i><b>7.7</b> Resource Forms</a></li>
<li class="chapter" data-level="7.8" data-path="resourcer.html"><a href="resourcer.html#examples-of-resources"><i class="fa fa-check"></i><b>7.8</b> Examples of Resources</a></li>
<li class="chapter" data-level="7.9" data-path="resourcer.html"><a href="resourcer.html#using-resources-with-datashield"><i class="fa fa-check"></i><b>7.9</b> Using Resources with DataSHIELD</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="resourcer.html"><a href="resourcer.html#tsv-file-into-a-tibble-or-data.frame"><i class="fa fa-check"></i><b>7.9.1</b> TSV File into a tibble or data.frame</a></li>
<li class="chapter" data-level="7.9.2" data-path="resourcer.html"><a href="resourcer.html#r-data-file-into-a-r-object"><i class="fa fa-check"></i><b>7.9.2</b> R Data File into a R object</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>III Resources Extensions</b></span></li>
<li class="chapter" data-level="8" data-path="omic-extension.html"><a href="omic-extension.html"><i class="fa fa-check"></i><b>8</b> Extension to VCF files to peform GWAS with Bioconductor</a></li>
<li class="chapter" data-level="9" data-path="shell-extension.html"><a href="shell-extension.html"><i class="fa fa-check"></i><b>9</b> Extension to secure shell programs: GWAS with PLINK</a></li>
<li class="chapter" data-level="10" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html"><i class="fa fa-check"></i><b>10</b> Extension to GA4GH and EGA (BAM/VCF files)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#upload-ga4gh-and-ega-resources-into-opal"><i class="fa fa-check"></i><b>10.1</b> Upload GA4GH and EGA resources into OPAL</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#manually"><i class="fa fa-check"></i><b>10.1.1</b> Manually</a></li>
<li class="chapter" data-level="10.1.2" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#using-r"><i class="fa fa-check"></i><b>10.1.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#adding-a-genomic-resource"><i class="fa fa-check"></i><b>10.2</b> Adding a genomic resource</a></li>
<li class="chapter" data-level="10.3" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#descriptive-omic-data-analysis"><i class="fa fa-check"></i><b>10.3</b> Descriptive omic data analysis</a></li>
</ul></li>
<li class="part"><span><b>IV Workflows</b></span></li>
<li class="chapter" data-level="11" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>11</b> Setup</a>
<ul>
<li class="chapter" data-level="11.1" data-path="setup.html"><a href="setup.html#providing-datashield-packages-in-the-opal-server"><i class="fa fa-check"></i><b>11.1</b> Providing DataSHIELD packages in the opal server</a></li>
<li class="chapter" data-level="11.2" data-path="setup.html"><a href="setup.html#required-r-packages-in-the-client-site-e.g.-local-machine"><i class="fa fa-check"></i><b>11.2</b> Required R Packages in the client site (e.g. local machine)</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="basic-statistical-analyses.html"><a href="basic-statistical-analyses.html"><i class="fa fa-check"></i><b>12</b> Basic statistical analyses</a>
<ul>
<li class="chapter" data-level="12.1" data-path="basic-statistical-analyses.html"><a href="basic-statistical-analyses.html#analysis-from-a-single-study"><i class="fa fa-check"></i><b>12.1</b> Analysis from a single study</a></li>
<li class="chapter" data-level="12.2" data-path="basic-statistical-analyses.html"><a href="basic-statistical-analyses.html#analysis-from-a-multiple-studies"><i class="fa fa-check"></i><b>12.2</b> Analysis from a multiple studies</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Omic.html"><a href="Omic.html"><i class="fa fa-check"></i><b>13</b> Omic data analysis: types of implemented analyses</a></li>
<li class="chapter" data-level="14" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html"><i class="fa fa-check"></i><b>14</b> Differential gene expression (DGE) analysis</a></li>
<li class="chapter" data-level="15" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html"><i class="fa fa-check"></i><b>15</b> Epigenome-wide association analysis (EWAS)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#single-cpg-analysis"><i class="fa fa-check"></i><b>15.1</b> Single CpG analysis</a></li>
<li class="chapter" data-level="15.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#multiple-cpg-analysis"><i class="fa fa-check"></i><b>15.2</b> Multiple CpG analysis</a></li>
<li class="chapter" data-level="15.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#adjusting-for-surrogate-variables"><i class="fa fa-check"></i><b>15.3</b> Adjusting for Surrogate Variables</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="genome-wide-association-analyses-gwas.html"><a href="genome-wide-association-analyses-gwas.html"><i class="fa fa-check"></i><b>16</b> Genome-wide association analyses (GWAS)</a>
<ul>
<li class="chapter" data-level="16.1" data-path="genome-wide-association-analyses-gwas.html"><a href="genome-wide-association-analyses-gwas.html#gwasBioc"><i class="fa fa-check"></i><b>16.1</b> GWAS with Bioconductor</a></li>
<li class="chapter" data-level="16.2" data-path="genome-wide-association-analyses-gwas.html"><a href="genome-wide-association-analyses-gwas.html#gwas-with-plink"><i class="fa fa-check"></i><b>16.2</b> GWAS with PLINK</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="GIS.html"><a href="GIS.html"><i class="fa fa-check"></i><b>17</b> Geospatial data analysis</a>
<ul>
<li class="chapter" data-level="17.1" data-path="GIS.html"><a href="GIS.html#introducing-the-analysis"><i class="fa fa-check"></i><b>17.1</b> Introducing the analysis</a></li>
<li class="chapter" data-level="17.2" data-path="GIS.html"><a href="GIS.html#setting-up-the-analysis"><i class="fa fa-check"></i><b>17.2</b> Setting up the analysis</a></li>
<li class="chapter" data-level="17.3" data-path="GIS.html"><a href="GIS.html#data-manipulation"><i class="fa fa-check"></i><b>17.3</b> Data manipulation</a></li>
<li class="chapter" data-level="17.4" data-path="GIS.html"><a href="GIS.html#generating-the-final-result"><i class="fa fa-check"></i><b>17.4</b> Generating the final result</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="exposome.html"><a href="exposome.html"><i class="fa fa-check"></i><b>18</b> Exposome data analysis</a>
<ul>
<li class="chapter" data-level="18.1" data-path="exposome.html"><a href="exposome.html#loading-the-exposome-set"><i class="fa fa-check"></i><b>18.1</b> Loading the Exposome Set</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="exposome.html"><a href="exposome.html#from-tables-to-exposomeset"><i class="fa fa-check"></i><b>18.1.1</b> From tables to <code>ExposomeSet</code></a></li>
<li class="chapter" data-level="18.1.2" data-path="exposome.html"><a href="exposome.html#from-resource-to-exposomeset"><i class="fa fa-check"></i><b>18.1.2</b> From resource to <code>ExposomeSet</code></a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="exposome.html"><a href="exposome.html#exploring-the-loaded-dataset"><i class="fa fa-check"></i><b>18.2</b> Exploring the loaded dataset</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="exposome.html"><a href="exposome.html#family-names"><i class="fa fa-check"></i><b>18.2.1</b> Family names</a></li>
<li class="chapter" data-level="18.2.2" data-path="exposome.html"><a href="exposome.html#exposures-and-phenotypes-names"><i class="fa fa-check"></i><b>18.2.2</b> Exposures and phenotypes names</a></li>
<li class="chapter" data-level="18.2.3" data-path="exposome.html"><a href="exposome.html#summary-of-variables"><i class="fa fa-check"></i><b>18.2.3</b> Summary of variables</a></li>
<li class="chapter" data-level="18.2.4" data-path="exposome.html"><a href="exposome.html#missing-data"><i class="fa fa-check"></i><b>18.2.4</b> Missing data</a></li>
<li class="chapter" data-level="18.2.5" data-path="exposome.html"><a href="exposome.html#exposures-normality"><i class="fa fa-check"></i><b>18.2.5</b> Exposures Normality</a></li>
<li class="chapter" data-level="18.2.6" data-path="exposome.html"><a href="exposome.html#exposures-behaviour"><i class="fa fa-check"></i><b>18.2.6</b> Exposures Behaviour</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="exposome.html"><a href="exposome.html#exposures-imputation"><i class="fa fa-check"></i><b>18.3</b> Exposures Imputation</a></li>
<li class="chapter" data-level="18.4" data-path="exposome.html"><a href="exposome.html#performing-an-exwas"><i class="fa fa-check"></i><b>18.4</b> Performing an ExWAS</a></li>
<li class="chapter" data-level="18.5" data-path="exposome.html"><a href="exposome.html#exposures-pca"><i class="fa fa-check"></i><b>18.5</b> Exposures PCA</a></li>
<li class="chapter" data-level="18.6" data-path="exposome.html"><a href="exposome.html#exposures-correlation"><i class="fa fa-check"></i><b>18.6</b> Exposures Correlation</a></li>
</ul></li>
<li class="part"><span><b>V Developers</b></span></li>
<li class="chapter" data-level="19" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html"><i class="fa fa-check"></i><b>19</b> DSLite: DataSHIELD Implementation on Local Datasets</a>
<ul>
<li class="chapter" data-level="19.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#development-environment-setup"><i class="fa fa-check"></i><b>19.1</b> Development Environment Setup</a>
<ul>
<li class="chapter" data-level="19.1.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-packages"><i class="fa fa-check"></i><b>19.1.1</b> DataSHIELD Packages</a></li>
<li class="chapter" data-level="19.1.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#test-datasets"><i class="fa fa-check"></i><b>19.1.2</b> Test Datasets</a></li>
<li class="chapter" data-level="19.1.3" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#r-package-development-tools"><i class="fa fa-check"></i><b>19.1.3</b> R Package Development Tools</a></li>
</ul></li>
<li class="chapter" data-level="19.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-development-flow"><i class="fa fa-check"></i><b>19.2</b> DataSHIELD Development Flow</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#dsliteserver"><i class="fa fa-check"></i><b>19.2.1</b> DSLiteServer</a></li>
<li class="chapter" data-level="19.2.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-configuration"><i class="fa fa-check"></i><b>19.2.2</b> DataSHIELD Configuration</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-sessions"><i class="fa fa-check"></i><b>19.3</b> DataSHIELD Sessions</a></li>
<li class="chapter" data-level="19.4" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#debugging"><i class="fa fa-check"></i><b>19.4</b> Debugging</a></li>
<li class="chapter" data-level="19.5" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#limitations"><i class="fa fa-check"></i><b>19.5</b> Limitations</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#function-parameters-parser"><i class="fa fa-check"></i><b>19.5.1</b> Function Parameters Parser</a></li>
<li class="chapter" data-level="19.5.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#server-side-environments"><i class="fa fa-check"></i><b>19.5.2</b> Server Side Environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="package-info.html"><a href="package-info.html"><i class="fa fa-check"></i><b>20</b> Creating DataSHIELD packages</a>
<ul>
<li class="chapter" data-level="20.1" data-path="package-info.html"><a href="package-info.html#example-of-datashield-package-development"><i class="fa fa-check"></i><b>20.1</b> Example of DataSHIELD package development</a>
<ul>
<li class="chapter" data-level="20.1.1" data-path="package-info.html"><a href="package-info.html#fandfiles"><i class="fa fa-check"></i><b>20.1.1</b> Folders and files</a></li>
<li class="chapter" data-level="20.1.2" data-path="package-info.html"><a href="package-info.html#development-of-a-function"><i class="fa fa-check"></i><b>20.1.2</b> Development of a function</a></li>
<li class="chapter" data-level="20.1.3" data-path="package-info.html"><a href="package-info.html#testing-of-the-function"><i class="fa fa-check"></i><b>20.1.3</b> Testing of the function</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html"><i class="fa fa-check"></i><b>21</b> Tips and tricks</a>
<ul>
<li class="chapter" data-level="21.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-create-a-new-project-into-opal"><i class="fa fa-check"></i><b>21.1</b> How to create a new project into OPAL</a>
<ul>
<li class="chapter" data-level="21.1.1" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#manually"><i class="fa fa-check"></i><b>21.1.1</b> Manually</a></li>
<li class="chapter" data-level="21.1.2" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#using-r"><i class="fa fa-check"></i><b>21.1.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-upload-a-new-resource-into-opal"><i class="fa fa-check"></i><b>21.2</b> How to upload a new resource into OPAL</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#create_manually"><i class="fa fa-check"></i><b>21.2.1</b> Manually</a></li>
<li class="chapter" data-level="21.2.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#using-r-1"><i class="fa fa-check"></i><b>21.2.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-install-datashield-packages-into-opal-server"><i class="fa fa-check"></i><b>21.3</b> How to install DataSHIELD packages into OPAL server</a>
<ul>
<li class="chapter" data-level="21.3.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#manually-1"><i class="fa fa-check"></i><b>21.3.1</b> Manually</a></li>
<li class="chapter" data-level="21.3.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#using-r-2"><i class="fa fa-check"></i><b>21.3.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-install-r-packages-into-opal-server"><i class="fa fa-check"></i><b>21.4</b> How to install R packages into OPAL server</a>
<ul>
<li class="chapter" data-level="21.4.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#manually-2"><i class="fa fa-check"></i><b>21.4.1</b> Manually</a></li>
<li class="chapter" data-level="21.4.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#using-r-3"><i class="fa fa-check"></i><b>21.4.2</b> Using R</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="advancedtechnical-questions.html"><a href="advancedtechnical-questions.html"><i class="fa fa-check"></i><b>22</b> Advanced/technical questions</a></li>
<li class="part"><span><b>VI Appendix</b></span></li>
<li class="chapter" data-level="23" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>23</b> Session Info</a></li>
<li class="chapter" data-level="24" data-path="contributors.html"><a href="contributors.html"><i class="fa fa-check"></i><b>24</b> Contributors</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exposome" class="section level1" number="18">
<h1><span class="header-section-number">18</span> Exposome data analysis</h1>
<div id="loading-the-exposome-set" class="section level2" number="18.1">
<h2><span class="header-section-number">18.1</span> Loading the Exposome Set</h2>
<p>An exposome dataset is made up from <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/rexposome/inst/doc/exposome_data_analysis.html#data-format">three different tables</a>. In order to be studied, this three tables need to be coerced into a R object of class <code>ExposomeSet</code>, for that reason an exposome dataset can be available on a server as three tables or as a resource.</p>
<div id="from-tables-to-exposomeset" class="section level3" number="18.1.1">
<h3><span class="header-section-number">18.1.1</span> From tables to <code>ExposomeSet</code></h3>
<p>To coerce the three tables that make the exposome dataset into an <code>ExposomeSet</code> R object, the tables have to be loaded inside the same study server. On this example, the tables are on the demo Opal server as resources, which means they have to be loaded on the study server and be coerced as data frames; for more information read the <a href="https://cran.r-project.org/web/packages/resourcer/index.html">resourcer</a> package documentation.</p>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="exposome.html#cb271-1"></a><span class="kw">library</span>(dsBaseClient)</span>
<span id="cb271-2"><a href="exposome.html#cb271-2"></a><span class="kw">library</span>(dsExposomeClient)</span>
<span id="cb271-3"><a href="exposome.html#cb271-3"></a><span class="kw">library</span>(DSOpal)</span>
<span id="cb271-4"><a href="exposome.html#cb271-4"></a></span>
<span id="cb271-5"><a href="exposome.html#cb271-5"></a>builder &lt;-<span class="st"> </span><span class="kw">newDSLoginBuilder</span>()</span>
<span id="cb271-6"><a href="exposome.html#cb271-6"></a>builder<span class="op">$</span><span class="kw">append</span>(<span class="dt">server =</span> <span class="st">&quot;server1&quot;</span>, <span class="dt">url =</span> <span class="st">&quot;https://opal-demo.obiba.org/&quot;</span>,</span>
<span id="cb271-7"><a href="exposome.html#cb271-7"></a>               <span class="dt">user =</span> <span class="st">&quot;administrator&quot;</span>, <span class="dt">password =</span> <span class="st">&quot;password&quot;</span>,</span>
<span id="cb271-8"><a href="exposome.html#cb271-8"></a>               <span class="dt">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>)</span>
<span id="cb271-9"><a href="exposome.html#cb271-9"></a>logindata &lt;-<span class="st"> </span>builder<span class="op">$</span><span class="kw">build</span>()</span>
<span id="cb271-10"><a href="exposome.html#cb271-10"></a>conns &lt;-<span class="st"> </span><span class="kw">datashield.login</span>(<span class="dt">logins =</span> logindata)</span>
<span id="cb271-11"><a href="exposome.html#cb271-11"></a></span>
<span id="cb271-12"><a href="exposome.html#cb271-12"></a><span class="kw">datashield.assign.resource</span>(conns, <span class="dt">symbol =</span> <span class="st">&#39;description&#39;</span>, <span class="dt">resource =</span> <span class="kw">list</span>(<span class="dt">server1 =</span> <span class="st">&#39;EXPOSOME.description&#39;</span>))</span>
<span id="cb271-13"><a href="exposome.html#cb271-13"></a><span class="kw">ds.class</span>(<span class="st">&#39;description&#39;</span>)</span></code></pre></div>
<pre><code>$server1
[1] &quot;TidyFileResourceClient&quot; &quot;FileResourceClient&quot;     &quot;ResourceClient&quot;         &quot;R6&quot;                    </code></pre>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="exposome.html#cb273-1"></a><span class="kw">datashield.assign.expr</span>(conns, <span class="dt">symbol =</span> <span class="st">&quot;description&quot;</span>, <span class="dt">expr =</span> <span class="kw">quote</span>(<span class="kw">as.resource.data.frame</span>(description)))</span>
<span id="cb273-2"><a href="exposome.html#cb273-2"></a><span class="kw">ds.class</span>(<span class="st">&#39;description&#39;</span>)</span></code></pre></div>
<pre><code>$server1
[1] &quot;spec_tbl_df&quot; &quot;tbl_df&quot;      &quot;tbl&quot;         &quot;data.frame&quot; </code></pre>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="exposome.html#cb275-1"></a><span class="kw">datashield.assign.resource</span>(conns, <span class="dt">symbol =</span> <span class="st">&#39;exposures&#39;</span>, <span class="dt">resource =</span> <span class="kw">list</span>(<span class="dt">server1 =</span> <span class="st">&#39;EXPOSOME.exposures&#39;</span>))</span>
<span id="cb275-2"><a href="exposome.html#cb275-2"></a><span class="kw">datashield.assign.expr</span>(conns, <span class="dt">symbol =</span> <span class="st">&quot;exposures&quot;</span>, <span class="dt">expr =</span> <span class="kw">quote</span>(<span class="kw">as.resource.data.frame</span>(exposures)))</span>
<span id="cb275-3"><a href="exposome.html#cb275-3"></a></span>
<span id="cb275-4"><a href="exposome.html#cb275-4"></a><span class="kw">datashield.assign.resource</span>(conns, <span class="dt">symbol =</span> <span class="st">&#39;phenotypes&#39;</span>, <span class="dt">resource =</span> <span class="kw">list</span>(<span class="dt">server1 =</span> <span class="st">&#39;EXPOSOME.phenotypes&#39;</span>))</span>
<span id="cb275-5"><a href="exposome.html#cb275-5"></a><span class="kw">datashield.assign.expr</span>(conns, <span class="dt">symbol =</span> <span class="st">&quot;phenotypes&quot;</span>, <span class="dt">expr =</span> <span class="kw">quote</span>(<span class="kw">as.resource.data.frame</span>(phenotypes)))</span>
<span id="cb275-6"><a href="exposome.html#cb275-6"></a></span>
<span id="cb275-7"><a href="exposome.html#cb275-7"></a><span class="kw">ds.ls</span>()</span></code></pre></div>
<pre><code>$server1
$server1$environment.searched
[1] &quot;R_GlobalEnv&quot;

$server1$objects.found
[1] &quot;description&quot; &quot;exposures&quot;   &quot;phenotypes&quot; </code></pre>
<p>If the tables are on the Opal server as tables instead of resources, the lines after the <code>datashield.login</code> would look like this</p>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="exposome.html#cb277-1"></a><span class="co"># Don&#39;t run this code, the tables are not on the demo Opal server, it&#39;s just for demostration purposes</span></span>
<span id="cb277-2"><a href="exposome.html#cb277-2"></a><span class="kw">datashield.assign.table</span>(conns, <span class="dt">symbol =</span> <span class="st">&#39;description&#39;</span>, <span class="dt">table =</span> <span class="kw">list</span>(<span class="dt">server1 =</span> <span class="st">&#39;EXPOSOME.description_table&#39;</span>))</span>
<span id="cb277-3"><a href="exposome.html#cb277-3"></a></span>
<span id="cb277-4"><a href="exposome.html#cb277-4"></a><span class="co"># Repeat for the other two tables</span></span></code></pre></div>
<p>When there are tables on the Opal server they can be directly loaded to the study server as data frames, there is no need to perform the <code>as.data.frame</code> assignation.</p>
<p>To coerce the three tables to the <code>ExposomeSet</code> object, the <code>dsExposomeClient</code> library has the <code>ds.loadExposome()</code> function, which takes as input parameters the names of the tables on the study server among some further configuration options, refer to the function documentation for more information.</p>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="exposome.html#cb278-1"></a><span class="kw">ds.loadExposome</span>(<span class="st">&quot;exposures&quot;</span>, <span class="st">&quot;description&quot;</span>, <span class="st">&quot;phenotypes&quot;</span>, <span class="st">&quot;idnum&quot;</span>, <span class="st">&quot;idnum&quot;</span>, <span class="st">&quot;Exposure&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="dv">5</span>, <span class="ot">FALSE</span>, <span class="st">&quot;exposome_object&quot;</span>)</span>
<span id="cb278-2"><a href="exposome.html#cb278-2"></a><span class="kw">ds.class</span>(<span class="st">&quot;exposome_object&quot;</span>)</span></code></pre></div>
<pre><code>$server1
[1] &quot;ExposomeSet&quot;
attr(,&quot;package&quot;)
[1] &quot;rexposome&quot;</code></pre>
</div>
<div id="from-resource-to-exposomeset" class="section level3" number="18.1.2">
<h3><span class="header-section-number">18.1.2</span> From resource to <code>ExposomeSet</code></h3>
<p>When there’s an <code>ExposomeSet</code> resource available, it just needs to be loaded into the study server</p>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb280-1"><a href="exposome.html#cb280-1"></a><span class="kw">datashield.assign.resource</span>(conns, <span class="dt">symbol =</span> <span class="st">&#39;exposome_resource&#39;</span>, <span class="dt">resource =</span> <span class="kw">list</span>(<span class="dt">server1 =</span> <span class="st">&#39;EXPOSOME.exposomeSet&#39;</span>))</span>
<span id="cb280-2"><a href="exposome.html#cb280-2"></a></span>
<span id="cb280-3"><a href="exposome.html#cb280-3"></a><span class="kw">ds.ls</span>()</span></code></pre></div>
<pre><code>$server1
$server1$environment.searched
[1] &quot;R_GlobalEnv&quot;

$server1$objects.found
[1] &quot;description&quot;       &quot;exposome_object&quot;   &quot;exposome_resource&quot; &quot;exposures&quot;         &quot;phenotypes&quot;       </code></pre>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="exposome.html#cb282-1"></a><span class="kw">ds.class</span>(<span class="st">&quot;exposome_resource&quot;</span>)</span></code></pre></div>
<pre><code>$server1
[1] &quot;RDataFileResourceClient&quot; &quot;FileResourceClient&quot;      &quot;ResourceClient&quot;          &quot;R6&quot;                     </code></pre>
<p>Once on the study server, the resource has to be coerced into an R object</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="exposome.html#cb284-1"></a><span class="kw">datashield.assign.expr</span>(conns, <span class="dt">symbol =</span> <span class="st">&quot;exposome_resource&quot;</span>, <span class="dt">expr =</span> <span class="kw">quote</span>(<span class="kw">as.resource.object</span>(exposome_resource)))</span>
<span id="cb284-2"><a href="exposome.html#cb284-2"></a><span class="kw">ds.class</span>(<span class="st">&quot;exposome_resource&quot;</span>)</span></code></pre></div>
<pre><code>$server1
[1] &quot;ExposomeSet&quot;
attr(,&quot;package&quot;)
[1] &quot;rexposome&quot;</code></pre>
</div>
</div>
<div id="exploring-the-loaded-dataset" class="section level2" number="18.2">
<h2><span class="header-section-number">18.2</span> Exploring the loaded dataset</h2>
<p>Once the Exposome datasets are loaded on the study servers, some simple functions can be used to have some understanding of what the dataset contains. Those correspond to the name of the exposures, the names of the families of the exposures and the name of the phenotypes.</p>
<div id="family-names" class="section level3" number="18.2.1">
<h3><span class="header-section-number">18.2.1</span> Family names</h3>
<p>To get the names of the families of exposures present on the exposome set, there’s the <code>ds.familyNames</code> function, it can take the parameter <code>by.exposure = TRUE</code> to list the family names and the exposures that contain each family.</p>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="exposome.html#cb286-1"></a><span class="kw">ds.familyNames</span>(<span class="st">&quot;exposome_object&quot;</span>)</span></code></pre></div>
<pre><code>$server1
 [1] &quot;Air Pollutants&quot;    &quot;Metals&quot;            &quot;PBDEs&quot;             &quot;Organochlorines&quot;   &quot;Bisphenol A&quot;      
 [6] &quot;Water Pollutants&quot;  &quot;Built Environment&quot; &quot;Cotinine&quot;          &quot;Home Environment&quot;  &quot;Phthalates&quot;       
[11] &quot;Noise&quot;             &quot;PFOAs&quot;             &quot;Temperature&quot;      </code></pre>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="exposome.html#cb288-1"></a><span class="kw">head</span>(<span class="kw">ds.familyNames</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="ot">TRUE</span>)<span class="op">$</span>server1)</span></code></pre></div>
<pre><code>         AbsPM25               As           BDE100           BDE138           BDE153           BDE154 
&quot;Air Pollutants&quot;         &quot;Metals&quot;          &quot;PBDEs&quot;          &quot;PBDEs&quot;          &quot;PBDEs&quot;          &quot;PBDEs&quot; </code></pre>
</div>
<div id="exposures-and-phenotypes-names" class="section level3" number="18.2.2">
<h3><span class="header-section-number">18.2.2</span> Exposures and phenotypes names</h3>
<p>To get the names of the exposures or phenotypes (or both combined) there’s the function <code>ds.exposome_variables</code>, it takes as argument <code>exposures</code>, <code>phenotypes</code> or <code>all</code> to retrieve the desired names.</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="exposome.html#cb290-1"></a><span class="kw">head</span>(<span class="kw">ds.exposome_variables</span>(<span class="st">&quot;exposome_object&quot;</span> , <span class="st">&quot;exposures&quot;</span>, conns)<span class="op">$</span>server1)</span></code></pre></div>
<pre><code>[1] &quot;AbsPM25&quot; &quot;As&quot;      &quot;BDE100&quot;  &quot;BDE138&quot;  &quot;BDE153&quot;  &quot;BDE154&quot; </code></pre>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="exposome.html#cb292-1"></a><span class="kw">ds.exposome_variables</span>(<span class="st">&quot;exposome_object&quot;</span> , <span class="st">&quot;phenotypes&quot;</span>, conns)</span></code></pre></div>
<pre><code>$server1
[1] &quot;whistling_chest&quot; &quot;flu&quot;             &quot;rhinitis&quot;        &quot;wheezing&quot;        &quot;birthdate&quot;       &quot;sex&quot;            
[7] &quot;age&quot;             &quot;cbmi&quot;            &quot;blood_pre&quot;      </code></pre>
</div>
<div id="summary-of-variables" class="section level3" number="18.2.3">
<h3><span class="header-section-number">18.2.3</span> Summary of variables</h3>
<p>Non-disclosive descriptive statistics can be obtained from the exposome dataset, the function <code>ds.exposome_summary</code> is in charge of that. It can obtain descriptive statistics from numeric and factor variables of the exposome dataset (both from exposures and phenotypes).</p>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="exposome.html#cb294-1"></a><span class="co"># Numerical variable</span></span>
<span id="cb294-2"><a href="exposome.html#cb294-2"></a><span class="kw">ds.exposome_summary</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="st">&quot;PM25&quot;</span>, conns)</span></code></pre></div>
<pre><code>$server1
$server1$class
[1] &quot;numeric&quot;

$server1$length
[1] 109

$server1$`quantiles &amp; mean`
      5%      10%      25%      50%      75%      90%      95%     Mean 
1.093217 1.132021 1.146654 1.171904 1.181451 1.202192 1.216758 1.165488 </code></pre>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="exposome.html#cb296-1"></a><span class="co"># Factor variable</span></span>
<span id="cb296-2"><a href="exposome.html#cb296-2"></a><span class="kw">ds.exposome_summary</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="st">&quot;flu&quot;</span>, conns)</span></code></pre></div>
<pre><code>$server1
$server1$class
[1] &quot;character&quot;

$server1$length
[1] 109</code></pre>
</div>
<div id="missing-data" class="section level3" number="18.2.4">
<h3><span class="header-section-number">18.2.4</span> Missing data</h3>
<p>The number of missing data on each exposure and on each phenotype can be found by using the function <code>ds.tableMissings</code>. This function returns a vector with the amount of missing data in each exposure or phenotype. The argument <code>set</code> indicates if the number of missing values is counted on exposures or phenotypes. The argument <code>output</code> indicates if it is shown as counts (<code>output="n"</code>) or as percentage (<code>output="p"</code>).</p>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="exposome.html#cb298-1"></a><span class="kw">ds.tableMissings</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="dt">set =</span> <span class="st">&quot;exposures&quot;</span>)</span></code></pre></div>
<pre><code>$server1
        Dens         Temp         Conn      AbsPM25           NO          NO2          NOx         PM10 
           0            0            1            2            2            2            2            2 
      PM10Cu       PM10Fe        PM10K       PM10Ni        PM10S       PM10SI       PM10Zn         PM25 
           2            2            2            2            2            2            2            2 
      PM25CU       PM25FE        PM25K       PM25Ni        PM25S       PM25Sl       PM25Zn     PMcoarse 
           2            2            2            2            2            2            2            2 
     Benzene        PM25V          ETS G_pesticides          Gas         BTHM        CHCl3 H_pesticides 
           3            3            5            5            5            6            6            6 
     Noise_d      Noise_n          THM     Cotinine         bHCH          DDE          DDT          HCB 
           6            6            6            7           13           13           13           13 
      PCB118       PCB138       PCB153       PCB180          BPA           As           Cs           Mo 
          13           13           13           13           21           24           24           24 
          Ni           Tl           Zn           Hg           Cd           Sb        Green           Cu 
          24           24           24           27           28           30           31           40 
       PM10V           Se         MBzP        MEHHP         MEHP        MEOHP          MEP         MiBP 
          41           45           46           46           46           46           46           46 
        MnBP     X5cxMEPP           Co        PFHxS         PFNA         PFOA         PFOS    X7OHMMeOP 
          46           46           47           48           48           48           48           49 
          Pb     X2cxMMHP       BDE100       BDE138       BDE153       BDE154        BDE17       BDE183 
          59           64           76           76           76           76           76           76 
      BDE190       BDE209        BDE28        BDE47        BDE66        BDE71        BDE85        BDE99 
          76           76           76           76           76           76           76           76 </code></pre>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="exposome.html#cb300-1"></a><span class="kw">ds.tableMissings</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="dt">set =</span> <span class="st">&quot;phenotypes&quot;</span>)</span></code></pre></div>
<pre><code>$server1
whistling_chest             flu        rhinitis        wheezing             sex             age 
              0               0               0               0               0               0 
           cbmi       blood_pre       birthdate 
              0               2               3 </code></pre>
<p>Optionally, there’s also the <code>ds.plotMissings</code> function which returns a <code>ggplot</code> object with a barplot of missings for exposures or phenotypes, there’s the option of displaying the percentage of missings as well as total counts, check the function documentation for further information.</p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="exposome.html#cb302-1"></a><span class="kw">ds.plotMissings</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="st">&quot;exposures&quot;</span>, <span class="st">&quot;p&quot;</span>)</span></code></pre></div>
<pre><code>$server1</code></pre>
<p><img src="fig/unnamed-chunk-74-1.png" width="672" /></p>
<p>Please note that since this function call returns a <code>ggplot</code> object, it can be plotted properly to avoid squished Y axis when there are lots of exposures, read the <a href="https://ggplot2.tidyverse.org/reference/ggsave.html">official documentation</a> for information on how to do that.</p>
</div>
<div id="exposures-normality" class="section level3" number="18.2.5">
<h3><span class="header-section-number">18.2.5</span> Exposures Normality</h3>
<p>Most of the test done in <em>exposome</em> analysis requires that the exposures must follow a normal distribution. The function <code>ds.normalityTest</code> performs a test on each exposure for normality behaviour. The result is a <code>data.frame</code> with the exposures’ names, a flag <code>TRUE</code>/<code>FALSE</code> for normality and the p-value obtained from the <em>Shapiro-Wilk</em> Normality Test (if the p-value is under the threshold, then the exposure is not normal).</p>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="exposome.html#cb304-1"></a>nm &lt;-<span class="st"> </span><span class="kw">ds.normalityTest</span>(<span class="st">&quot;exposome_object&quot;</span>)</span>
<span id="cb304-2"><a href="exposome.html#cb304-2"></a><span class="kw">table</span>(nm<span class="op">$</span>server1<span class="op">$</span>normality)</span></code></pre></div>
<pre><code>
FALSE  TRUE 
   55    29 </code></pre>
<p>So, the exposures that do not follow a normal distribution are:</p>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="exposome.html#cb306-1"></a>nm<span class="op">$</span>server1<span class="op">$</span>exposure[<span class="op">!</span>nm<span class="op">$</span>server1<span class="op">$</span>normality]</span></code></pre></div>
<pre><code> [1] &quot;DDT&quot;      &quot;PM10SI&quot;   &quot;PM25K&quot;    &quot;PM25Sl&quot;   &quot;PCB118&quot;   &quot;Tl&quot;       &quot;PM10V&quot;    &quot;PM25Zn&quot;   &quot;PM25FE&quot;  
[10] &quot;PM10K&quot;    &quot;BDE17&quot;    &quot;PM25&quot;     &quot;PMcoarse&quot; &quot;PM10&quot;     &quot;BPA&quot;      &quot;Green&quot;    &quot;NO2&quot;      &quot;Cs&quot;      
[19] &quot;PFNA&quot;     &quot;PCB153&quot;   &quot;PM25CU&quot;   &quot;MEOHP&quot;    &quot;Cu&quot;       &quot;HCB&quot;      &quot;MEHHP&quot;    &quot;DDE&quot;      &quot;BDE190&quot;  
[28] &quot;bHCH&quot;     &quot;PM10Zn&quot;   &quot;MnBP&quot;     &quot;NO&quot;       &quot;NOx&quot;      &quot;PM10S&quot;    &quot;MEHP&quot;     &quot;PCB138&quot;   &quot;Zn&quot;      
[37] &quot;X2cxMMHP&quot; &quot;PCB180&quot;   &quot;PFOA&quot;     &quot;Cotinine&quot; &quot;PM25S&quot;    &quot;Co&quot;       &quot;Conn&quot;     &quot;PM25Ni&quot;   &quot;PFHxS&quot;   
[46] &quot;PM10Ni&quot;   &quot;Cd&quot;       &quot;Dens&quot;     &quot;Se&quot;       &quot;X5cxMEPP&quot; &quot;BDE183&quot;   &quot;BDE28&quot;    &quot;Sb&quot;       &quot;BDE138&quot;  
[55] &quot;PM25V&quot;   </code></pre>
<p>The <code>ds.normalityTest</code> function has some extra input arguments to tune the normality test, check the function documentation for further information.</p>
<p>The exposures can be visualized using non-disclosive histograms to see their distribution along their <em>Shapiro-Wilk</em> Normality Test p-value.</p>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="exposome.html#cb308-1"></a><span class="kw">ds.exposure_histogram</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="st">&quot;BDE209&quot;</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-77-1.png" width="672" /></p>
<p>If the selected exposure is not following a normal distribution, the function <code>ds.exposure_histogram</code> accepts the argument <code>show.trans</code> to visualize the raw data histogram plus three typical transformations (<code>exp</code>, <code>log</code> and <code>sqrt</code>), the <em>Shapiro-Wilk</em> Normality Test p-value is shown for all the transformations.</p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="exposome.html#cb309-1"></a><span class="kw">ds.exposure_histogram</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="st">&quot;BDE209&quot;</span>, <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-78-1.png" width="672" /></p>
</div>
<div id="exposures-behaviour" class="section level3" number="18.2.6">
<h3><span class="header-section-number">18.2.6</span> Exposures Behaviour</h3>
<p>We can get a snapshot of the behaviour of the full exposome using the <code>ds.plotFamily</code> function. This function draws a mosaic of boxplots with all the numeric families of exposures, it can also plot single families and perform grouping inside of them. This function makes use of the <code>ggplot2</code> library to generate the plots. Future plans include displaying non-disclosive plots for non-numeric families.</p>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="exposome.html#cb310-1"></a><span class="kw">ds.plotFamily</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="dt">family =</span> <span class="st">&quot;all&quot;</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-79-1.png" width="672" /></p>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="exposome.html#cb311-1"></a><span class="kw">ds.plotFamily</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="dt">family =</span> <span class="st">&quot;Phthalates&quot;</span>, <span class="dt">group =</span> <span class="st">&quot;sex&quot;</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-79-2.png" width="672" /></p>
</div>
</div>
<div id="exposures-imputation" class="section level2" number="18.3">
<h2><span class="header-section-number">18.3</span> Exposures Imputation</h2>
<p>The missings of an exposome dataset can be imputed using the <code>ds.imputation</code> function, which calls the rexposome function <code>imputation</code> on the study server. Refer to the documentation of rexposome for details about the imputation procedures.</p>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="exposome.html#cb312-1"></a><span class="kw">ds.imputation</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="st">&quot;exposome_object_imputed&quot;</span>)</span></code></pre></div>
</div>
<div id="performing-an-exwas" class="section level2" number="18.4">
<h2><span class="header-section-number">18.4</span> Performing an ExWAS</h2>
<p>When the <code>ExposomeSet</code> object is on the study the server, the function <code>ds.exwas</code> is in charge of performing the ExWAS. As an example let’s present a situation where each exposition has to be associated to the <code>blood_pre</code> phenotype using the phenotype <code>sex</code> as a covariate. This study would be done as follows</p>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="exposome.html#cb313-1"></a>exwas_results &lt;-<span class="st"> </span><span class="kw">ds.exwas</span>(<span class="st">&quot;blood_pre ~ sex&quot;</span>, <span class="st">&quot;exposome_object&quot;</span>, <span class="st">&quot;gaussian&quot;</span>)</span>
<span id="cb313-2"><a href="exposome.html#cb313-2"></a><span class="kw">head</span>(exwas_results<span class="op">$</span>exwas_results)</span></code></pre></div>
<pre><code>  exposure         family coefficient      minE      maxE      p.value
1  AbsPM25 Air Pollutants  20.2567443 11.532500 28.980989 5.343657e-06
2       As         Metals   0.5049329 -1.637091  2.646956 6.440695e-01
3   BDE100          PBDEs  -1.4684745 -5.073223  2.136273 4.246181e-01
4   BDE138          PBDEs   1.2819339 -4.872401  7.436269 6.830857e-01
5   BDE153          PBDEs   2.8699183 -1.252769  6.992606 1.724462e-01
6   BDE154          PBDEs   0.1317781 -5.381815  5.645371 9.626373e-01</code></pre>
<p>The model is inputted as a string, where the left side term is the objective phenotype, and the right term are the covariates, in the case of wanting more covariates proceed as: <code>objective ~ cov1 + cov2 + ... + covN</code>. It’s important noting that if the output family (third argument) does not match the objective family, the ExWAS will fail (Example: The objective family is <code>binomial</code> and the user sets it to <code>gaussian</code>).</p>
<p>To visualize the results from the ExWAS, the function <code>ds.plotExwas</code> takes the output of <code>ds.exwas</code> and creates two different visualizations. A Manhattan-like plot with the p-values of the association between each exposure and asthma, coloured by families of exposures:</p>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="exposome.html#cb315-1"></a><span class="kw">ds.plotExwas</span>(exwas_results, <span class="st">&quot;manhattan&quot;</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-82-1.png" width="960" /></p>
<p>Also, a plot for the effects of a given model can be obtained with:</p>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb316-1"><a href="exposome.html#cb316-1"></a><span class="kw">ds.plotExwas</span>(exwas_results, <span class="st">&quot;effect&quot;</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-83-1.png" width="960" /></p>
</div>
<div id="exposures-pca" class="section level2" number="18.5">
<h2><span class="header-section-number">18.5</span> Exposures PCA</h2>
<p>A PCA can be performed on the exposures of the exposome dataset. To do so, there’s the <code>ds.exposome_pca</code>. The exposures should be standardized in order to perform the PCA properly, to do so, there’s the arguments <code>standar</code> and <code>method</code>, which standardize the Exposome Set before performing the PCA following the specified <code>method</code>. The available methods are <code>normal</code> (default method), which scales the exposures using the mean as the center and the standard variation as dispersion; the <code>robust</code> method, which uses the median and median absolute deviation respectively; and, <code>interquartile range</code>, which uses the median as the center and the coeficient between the interquartile range of the exposure and the normal range between the percentile 75 and 25 as variance. It is important noting that this function is sensitive to be disclosive, specially for very rectangular data frames (similar number of variables as individuals). To illustrate this problem, let’s try to perform a PCA on the whole exposures test data.</p>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb317-1"><a href="exposome.html#cb317-1"></a><span class="kw">ds.exposome_pca</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="dt">standar =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>Error: There are some DataSHIELD errors, list them with datashield.errors()</code></pre>
<p>If that is the case, one option is to reduce the families of exposures of the Exposome Set. The <code>ds.exposome_pca</code> function has the argument <code>fam</code> to select the families to subset the Exposome Set to perform the PCA.</p>
<div class="sourceCode" id="cb319"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb319-1"><a href="exposome.html#cb319-1"></a><span class="kw">ds.exposome_pca</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="dt">fam =</span> <span class="kw">c</span>(<span class="st">&quot;Metals&quot;</span>, <span class="st">&quot;Noise&quot;</span>), <span class="dt">standar =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p>The PCA function saves the results on the study server to prevent any dislosures, the default variable they take is called <code>"ds.exposome_pca.Results"</code>, which has to be passed to the visualization function. To visualize the results of the PCA there is the function <code>ds.exposome_pca_plot</code>, this function relies on the visualization methods already implemented on <code>rexposome</code> for the PCA analysis, it does it however on a non-disclosive way, by passing the scatter plot points through an anonimization process, hence the arguments <code>k</code>, <code>method</code> and <code>noise</code>. The visualization is controlled with the <code>set</code> argument, which takes <code>"all"</code> (mosaic of plots of the PCA), <code>"exposures"</code> (plot of the exposures space on the first two principal components, color coded by family), <code>"samples"</code> (plot of the individuals space on the first two principal components, this plot can take the <code>phenotype</code> argument to color code the individuals by phenotypes), <code>"variance"</code> and <code>"variance_explained"</code>, the two variance plots are quite self explanatory, the color code on the <code>"variance"</code> highlights the first two principal components as they are the ones shown on the other drawings.</p>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb320-1"><a href="exposome.html#cb320-1"></a><span class="kw">ds.exposome_pca_plot</span>(<span class="st">&quot;ds.exposome_pca.Results&quot;</span>, <span class="dt">set =</span> <span class="st">&quot;all&quot;</span>, <span class="dt">method =</span> <span class="dv">1</span>, <span class="dt">k =</span> <span class="dv">3</span>, <span class="dt">noise =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-86-1.png" width="672" /></p>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb321-1"><a href="exposome.html#cb321-1"></a><span class="kw">ds.exposome_pca_plot</span>(<span class="st">&quot;ds.exposome_pca.Results&quot;</span>, <span class="dt">set =</span> <span class="st">&quot;samples&quot;</span>, <span class="dt">phenotype =</span> <span class="st">&quot;sex&quot;</span>, <span class="dt">method =</span> <span class="dv">1</span>, <span class="dt">k =</span> <span class="dv">3</span>, <span class="dt">noise =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-86-2.png" width="672" /></p>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb322-1"><a href="exposome.html#cb322-1"></a><span class="kw">ds.exposome_pca_plot</span>(<span class="st">&quot;ds.exposome_pca.Results&quot;</span>, <span class="dt">set =</span> <span class="st">&quot;exposures&quot;</span>, <span class="dt">method =</span> <span class="dv">1</span>, <span class="dt">k =</span> <span class="dv">3</span>, <span class="dt">noise =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-86-3.png" width="672" /></p>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="exposome.html#cb323-1"></a><span class="kw">ds.exposome_pca_plot</span>(<span class="st">&quot;ds.exposome_pca.Results&quot;</span>, <span class="dt">set =</span> <span class="st">&quot;variance&quot;</span>, <span class="dt">method =</span> <span class="dv">1</span>, <span class="dt">k =</span> <span class="dv">3</span>, <span class="dt">noise =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-86-4.png" width="672" /></p>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="exposome.html#cb324-1"></a><span class="kw">ds.exposome_pca_plot</span>(<span class="st">&quot;ds.exposome_pca.Results&quot;</span>, <span class="dt">set =</span> <span class="st">&quot;variance_explained&quot;</span>, <span class="dt">method =</span> <span class="dv">1</span>, <span class="dt">k =</span> <span class="dv">3</span>, <span class="dt">noise =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-86-5.png" width="672" /></p>
<p>Furthermore, the <code>ds.exposome_pca_plot</code> function can plot the correlations betweeen the principal components and the exposures and the association of the phenotypes with the principal components. This two visualizations can be obtained by setting the <code>set</code> argument to <code>"exposures_correlation"</code> and <code>"phenotypes_correlation"</code> respectively.</p>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="exposome.html#cb325-1"></a><span class="kw">ds.exposome_pca_plot</span>(<span class="st">&quot;ds.exposome_pca.Results&quot;</span>, <span class="dt">set =</span> <span class="st">&quot;exposures_correlation&quot;</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-87-1.png" width="672" /></p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="exposome.html#cb326-1"></a><span class="kw">ds.exposome_pca_plot</span>(<span class="st">&quot;ds.exposome_pca.Results&quot;</span>, <span class="dt">set =</span> <span class="st">&quot;phenotypes_correlation&quot;</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-87-2.png" width="672" /></p>
</div>
<div id="exposures-correlation" class="section level2" number="18.6">
<h2><span class="header-section-number">18.6</span> Exposures Correlation</h2>
<p>The correlation between exposures, in terms of intra-family and inter-family exposures, is interesting to take into account. The correlation of the exposome can be computed using <code>ds.correlation</code>. The correlation could be disclosive, for that reason there’s the argument <code>fam</code> to select the families to compute the correlation subsetting the exposome dataset by families, typical complete exposome datasets will fail as the correlation matrix could be disclosive.</p>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb327-1"><a href="exposome.html#cb327-1"></a><span class="kw">ds.correlation</span>(<span class="st">&quot;exposome_object&quot;</span>)[[<span class="dv">1</span>]][[<span class="dv">1</span>]]<span class="op">$</span><span class="st">`</span><span class="dt">Correlation Matrix</span><span class="st">`</span>[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>     [,1] [,2] [,3] [,4] [,5]
[1,]   NA   NA   NA   NA   NA
[2,]   NA   NA   NA   NA   NA
[3,]   NA   NA   NA   NA   NA
[4,]   NA   NA   NA   NA   NA
[5,]   NA   NA   NA   NA   NA</code></pre>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb329-1"><a href="exposome.html#cb329-1"></a><span class="kw">ds.correlation</span>(<span class="st">&quot;exposome_object&quot;</span>)[[<span class="dv">1</span>]][[<span class="dv">1</span>]]<span class="op">$</span><span class="st">`</span><span class="dt">Error message</span><span class="st">`</span></span></code></pre></div>
<pre><code>[1] &quot;ERROR: The ratio of the number of variables over the number of individual-level records exceeds the allowed threshold, there is a possible risk of disclosure&quot;</code></pre>
<p>When subsetting the exposome set by families the correlation succeeds.</p>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="exposome.html#cb331-1"></a><span class="kw">ds.correlation</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="kw">c</span>(<span class="st">&quot;Metals&quot;</span>, <span class="st">&quot;Noise&quot;</span>))[[<span class="dv">1</span>]][[<span class="dv">1</span>]]<span class="op">$</span><span class="st">`</span><span class="dt">Correlation Matrix</span><span class="st">`</span>[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>            [,1]       [,2]        [,3]       [,4]        [,5]
[1,]  1.00000000 0.08897651 -0.12603679  0.1112793 -0.03127031
[2,]  0.08897651 1.00000000  0.10209881  0.5356812  0.84660044
[3,] -0.12603679 0.10209881  1.00000000 -0.5111215 -0.05851681
[4,]  0.11127929 0.53568121 -0.51112146  1.0000000  0.47229280
[5,] -0.03127031 0.84660044 -0.05851681  0.4722928  1.00000000</code></pre>
<p>The output from the <code>ds.correlation</code> consists of the correlation matrix and a vector that contains the names of exposures and families on the correlation table, this second item is used by the <code>corPlot</code> function to display the results of the correlation study.</p>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb333-1"><a href="exposome.html#cb333-1"></a>corr_data &lt;-<span class="st"> </span><span class="kw">ds.correlation</span>(<span class="st">&quot;exposome_object&quot;</span>, <span class="kw">c</span>(<span class="st">&quot;Metals&quot;</span>, <span class="st">&quot;Noise&quot;</span>))</span>
<span id="cb333-2"><a href="exposome.html#cb333-2"></a><span class="kw">corPlot</span>(corr_data)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-90-1.png" width="672" /></p>
<p>The best option to see the inter-family correlations is the <em>circos of correlations</em> while the <em>matrix of correlations</em> is a better way for studying the intra-family correlations.</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="exposome.html#cb334-1"></a><span class="kw">corPlot</span>(corr_data, <span class="dt">type =</span> <span class="st">&quot;circos&quot;</span>)</span></code></pre></div>
<p><img src="fig/unnamed-chunk-91-1.png" width="672" /></p>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="exposome.html#cb335-1"></a><span class="kw">datashield.logout</span>(conns)</span></code></pre></div>

</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="GIS.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="dslite-datashield-implementation-on-local-datasets.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
