<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>16 Genome-wide association analyses (GWAS) | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD</title>
  <meta name="description" content="16 Genome-wide association analyses (GWAS) | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="16 Genome-wide association analyses (GWAS) | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="16 Genome-wide association analyses (GWAS) | Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD" />
  
  
  



<meta name="date" content="2021-01-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="epigenome-wide-association-analysis-ewas.html"/>
<link rel="next" href="GIS.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="citation.html"><a href="citation.html"><i class="fa fa-check"></i>Citation</a></li>
<li class="part"><span><b>I Preamble</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#what-you-will-learn"><i class="fa fa-check"></i><b>1.1</b> What you will learn</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#preliminaries"><i class="fa fa-check"></i><b>1.2</b> Preliminaries</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#acknowledgments"><i class="fa fa-check"></i><b>1.3</b> Acknowledgments</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="learnR.html"><a href="learnR.html"><i class="fa fa-check"></i><b>2</b> Learning R</a></li>
<li class="chapter" data-level="3" data-path="BioC.html"><a href="BioC.html"><i class="fa fa-check"></i><b>3</b> Bioconductor Data Structures</a>
<ul>
<li class="chapter" data-level="3.1" data-path="BioC.html"><a href="BioC.html#snp-array-data"><i class="fa fa-check"></i><b>3.1</b> SNP Array Data</a></li>
<li class="chapter" data-level="3.2" data-path="BioC.html"><a href="BioC.html#expression-sets"><i class="fa fa-check"></i><b>3.2</b> Expression Sets</a></li>
<li class="chapter" data-level="3.3" data-path="BioC.html"><a href="BioC.html#genomic-ranges"><i class="fa fa-check"></i><b>3.3</b> Genomic Ranges</a></li>
<li class="chapter" data-level="3.4" data-path="BioC.html"><a href="BioC.html#summarized-experiments"><i class="fa fa-check"></i><b>3.4</b> Summarized Experiments</a></li>
<li class="chapter" data-level="3.5" data-path="BioC.html"><a href="BioC.html#ranged-summarized-experiments"><i class="fa fa-check"></i><b>3.5</b> Ranged Summarized Experiments</a></li>
<li class="chapter" data-level="3.6" data-path="BioC.html"><a href="BioC.html#multi-data-set"><i class="fa fa-check"></i><b>3.6</b> Multi Data Set</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="opal.html"><a href="opal.html"><i class="fa fa-check"></i><b>4</b> Opal</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction.html"><a href="introduction.html#introduction"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="opal.html"><a href="opal.html#data-management"><i class="fa fa-check"></i><b>4.2</b> Data Management</a></li>
<li class="chapter" data-level="4.3" data-path="opal.html"><a href="opal.html#security"><i class="fa fa-check"></i><b>4.3</b> Security</a></li>
<li class="chapter" data-level="4.4" data-path="opal.html"><a href="opal.html#r-integration"><i class="fa fa-check"></i><b>4.4</b> R Integration</a></li>
<li class="chapter" data-level="4.5" data-path="opal.html"><a href="opal.html#opal-demo-site"><i class="fa fa-check"></i><b>4.5</b> Opal demo site</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="datashield.html"><a href="datashield.html"><i class="fa fa-check"></i><b>5</b> DataSHIELD</a>
<ul>
<li class="chapter" data-level="5.1" data-path="introduction.html"><a href="introduction.html#introduction"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="datashield.html"><a href="datashield.html#datashield-r-interface-dsi"><i class="fa fa-check"></i><b>5.2</b> DataSHIELD R Interface (DSI)</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="datashield.html"><a href="datashield.html#class-structures"><i class="fa fa-check"></i><b>5.2.1</b> Class Structures</a></li>
<li class="chapter" data-level="5.2.2" data-path="datashield.html"><a href="datashield.html#higher-level-functions"><i class="fa fa-check"></i><b>5.2.2</b> Higher Level Functions</a></li>
<li class="chapter" data-level="5.2.3" data-path="datashield.html"><a href="datashield.html#options"><i class="fa fa-check"></i><b>5.2.3</b> Options</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="datashield.html"><a href="datashield.html#datashieldopal-implementation"><i class="fa fa-check"></i><b>5.3</b> DataSHIELD/Opal Implementation</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="datashield.html"><a href="datashield.html#client"><i class="fa fa-check"></i><b>5.3.1</b> Client</a></li>
<li class="chapter" data-level="5.3.2" data-path="datashield.html"><a href="datashield.html#server"><i class="fa fa-check"></i><b>5.3.2</b> Server</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="datashield.html"><a href="datashield.html#demo"><i class="fa fa-check"></i><b>5.4</b> Demo</a></li>
</ul></li>
<li class="part"><span><b>II Focus Topics</b></span></li>
<li class="chapter" data-level="6" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>6</b> The Resources</a>
<ul>
<li class="chapter" data-level="6.1" data-path="resources.html"><a href="resources.html#concept"><i class="fa fa-check"></i><b>6.1</b> Concept</a></li>
<li class="chapter" data-level="6.2" data-path="resources.html"><a href="resources.html#types"><i class="fa fa-check"></i><b>6.2</b> Types</a></li>
<li class="chapter" data-level="6.3" data-path="resources.html"><a href="resources.html#definition"><i class="fa fa-check"></i><b>6.3</b> Definition</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="resourcer.html"><a href="resourcer.html"><i class="fa fa-check"></i><b>7</b> The resourcer Package</a>
<ul>
<li class="chapter" data-level="7.1" data-path="introduction.html"><a href="introduction.html#introduction"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="resourcer.html"><a href="resourcer.html#file-resources"><i class="fa fa-check"></i><b>7.2</b> File Resources</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="resourcer.html"><a href="resourcer.html#file-getter"><i class="fa fa-check"></i><b>7.2.1</b> File Getter</a></li>
<li class="chapter" data-level="7.2.2" data-path="resourcer.html"><a href="resourcer.html#file-data-format"><i class="fa fa-check"></i><b>7.2.2</b> File Data Format</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="resourcer.html"><a href="resourcer.html#database-resources"><i class="fa fa-check"></i><b>7.3</b> Database Resources</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="resourcer.html"><a href="resourcer.html#dbi-connectors"><i class="fa fa-check"></i><b>7.3.1</b> DBI Connectors</a></li>
<li class="chapter" data-level="7.3.2" data-path="resourcer.html"><a href="resourcer.html#use-dplyr"><i class="fa fa-check"></i><b>7.3.2</b> Use dplyr</a></li>
<li class="chapter" data-level="7.3.3" data-path="resourcer.html"><a href="resourcer.html#document-databases"><i class="fa fa-check"></i><b>7.3.3</b> Document Databases</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="resourcer.html"><a href="resourcer.html#computation-resources"><i class="fa fa-check"></i><b>7.4</b> Computation Resources</a></li>
<li class="chapter" data-level="7.5" data-path="resourcer.html"><a href="resourcer.html#interacting-with-r-resources"><i class="fa fa-check"></i><b>7.5</b> Interacting with R Resources</a></li>
<li class="chapter" data-level="7.6" data-path="resourcer.html"><a href="resourcer.html#extending_resources"><i class="fa fa-check"></i><b>7.6</b> Extending Resources</a></li>
<li class="chapter" data-level="7.7" data-path="resourcer.html"><a href="resourcer.html#resource-forms"><i class="fa fa-check"></i><b>7.7</b> Resource Forms</a></li>
<li class="chapter" data-level="7.8" data-path="resourcer.html"><a href="resourcer.html#examples-of-resources"><i class="fa fa-check"></i><b>7.8</b> Examples of Resources</a></li>
<li class="chapter" data-level="7.9" data-path="resourcer.html"><a href="resourcer.html#using-resources-with-datashield"><i class="fa fa-check"></i><b>7.9</b> Using Resources with DataSHIELD</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="resourcer.html"><a href="resourcer.html#tsv-file-into-a-tibble-or-data.frame"><i class="fa fa-check"></i><b>7.9.1</b> TSV File into a tibble or data.frame</a></li>
<li class="chapter" data-level="7.9.2" data-path="resourcer.html"><a href="resourcer.html#r-data-file-into-a-r-object"><i class="fa fa-check"></i><b>7.9.2</b> R Data File into a R object</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>III Resources Extensions</b></span></li>
<li class="chapter" data-level="8" data-path="omic-extension.html"><a href="omic-extension.html"><i class="fa fa-check"></i><b>8</b> Extension to VCF files to peform GWAS with Bioconductor</a></li>
<li class="chapter" data-level="9" data-path="shell-extension.html"><a href="shell-extension.html"><i class="fa fa-check"></i><b>9</b> Extension to secure shell programs: GWAS with PLINK</a></li>
<li class="chapter" data-level="10" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html"><i class="fa fa-check"></i><b>10</b> Extension to GA4GH and EGA (BAM/VCF files)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#upload-ga4gh-and-ega-resources-into-opal"><i class="fa fa-check"></i><b>10.1</b> Upload GA4GH and EGA resources into OPAL</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#manually"><i class="fa fa-check"></i><b>10.1.1</b> Manually</a></li>
<li class="chapter" data-level="10.1.2" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#using-r"><i class="fa fa-check"></i><b>10.1.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#adding-a-genomic-resource"><i class="fa fa-check"></i><b>10.2</b> Adding a genomic resource</a></li>
<li class="chapter" data-level="10.3" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#descriptive-omic-data-analysis"><i class="fa fa-check"></i><b>10.3</b> Descriptive omic data analysis</a></li>
</ul></li>
<li class="part"><span><b>IV Workflows</b></span></li>
<li class="chapter" data-level="11" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>11</b> Setup</a>
<ul>
<li class="chapter" data-level="11.1" data-path="setup.html"><a href="setup.html#providing-datashield-packages-in-the-opal-server"><i class="fa fa-check"></i><b>11.1</b> Providing DataSHIELD packages in the opal server</a></li>
<li class="chapter" data-level="11.2" data-path="setup.html"><a href="setup.html#required-r-packages-in-the-client-site-e.g.-local-machine"><i class="fa fa-check"></i><b>11.2</b> Required R Packages in the client site (e.g. local machine)</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="basic-statistical-analyses.html"><a href="basic-statistical-analyses.html"><i class="fa fa-check"></i><b>12</b> Basic statistical analyses</a>
<ul>
<li class="chapter" data-level="12.1" data-path="basic-statistical-analyses.html"><a href="basic-statistical-analyses.html#analysis-from-a-single-study"><i class="fa fa-check"></i><b>12.1</b> Analysis from a single study</a></li>
<li class="chapter" data-level="12.2" data-path="basic-statistical-analyses.html"><a href="basic-statistical-analyses.html#analysis-from-a-multiple-studies"><i class="fa fa-check"></i><b>12.2</b> Analysis from a multiple studies</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Omic.html"><a href="Omic.html"><i class="fa fa-check"></i><b>13</b> Omic data analysis: types of implemented analyses</a></li>
<li class="chapter" data-level="14" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html"><i class="fa fa-check"></i><b>14</b> Differential gene expression (DGE) analysis</a></li>
<li class="chapter" data-level="15" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html"><i class="fa fa-check"></i><b>15</b> Epigenome-wide association analysis (EWAS)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#single-cpg-analysis"><i class="fa fa-check"></i><b>15.1</b> Single CpG analysis</a></li>
<li class="chapter" data-level="15.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#multiple-cpg-analysis"><i class="fa fa-check"></i><b>15.2</b> Multiple CpG analysis</a></li>
<li class="chapter" data-level="15.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#adjusting-for-surrogate-variables"><i class="fa fa-check"></i><b>15.3</b> Adjusting for Surrogate Variables</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="genome-wide-association-analyses-gwas.html"><a href="genome-wide-association-analyses-gwas.html"><i class="fa fa-check"></i><b>16</b> Genome-wide association analyses (GWAS)</a>
<ul>
<li class="chapter" data-level="16.1" data-path="genome-wide-association-analyses-gwas.html"><a href="genome-wide-association-analyses-gwas.html#gwasBioc"><i class="fa fa-check"></i><b>16.1</b> GWAS with Bioconductor</a></li>
<li class="chapter" data-level="16.2" data-path="genome-wide-association-analyses-gwas.html"><a href="genome-wide-association-analyses-gwas.html#gwas-with-plink"><i class="fa fa-check"></i><b>16.2</b> GWAS with PLINK</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="GIS.html"><a href="GIS.html"><i class="fa fa-check"></i><b>17</b> Geospatial data analysis</a>
<ul>
<li class="chapter" data-level="17.1" data-path="GIS.html"><a href="GIS.html#introducing-the-analysis"><i class="fa fa-check"></i><b>17.1</b> Introducing the analysis</a></li>
<li class="chapter" data-level="17.2" data-path="GIS.html"><a href="GIS.html#setting-up-the-analysis"><i class="fa fa-check"></i><b>17.2</b> Setting up the analysis</a></li>
<li class="chapter" data-level="17.3" data-path="GIS.html"><a href="GIS.html#data-manipulation"><i class="fa fa-check"></i><b>17.3</b> Data manipulation</a></li>
<li class="chapter" data-level="17.4" data-path="GIS.html"><a href="GIS.html#generating-the-final-result"><i class="fa fa-check"></i><b>17.4</b> Generating the final result</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="exposome.html"><a href="exposome.html"><i class="fa fa-check"></i><b>18</b> Exposome data analysis</a>
<ul>
<li class="chapter" data-level="18.1" data-path="exposome.html"><a href="exposome.html#loading-the-exposome-set"><i class="fa fa-check"></i><b>18.1</b> Loading the Exposome Set</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="exposome.html"><a href="exposome.html#from-tables-to-exposomeset"><i class="fa fa-check"></i><b>18.1.1</b> From tables to <code>ExposomeSet</code></a></li>
<li class="chapter" data-level="18.1.2" data-path="exposome.html"><a href="exposome.html#from-resource-to-exposomeset"><i class="fa fa-check"></i><b>18.1.2</b> From resource to <code>ExposomeSet</code></a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="exposome.html"><a href="exposome.html#exploring-the-loaded-dataset"><i class="fa fa-check"></i><b>18.2</b> Exploring the loaded dataset</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="exposome.html"><a href="exposome.html#family-names"><i class="fa fa-check"></i><b>18.2.1</b> Family names</a></li>
<li class="chapter" data-level="18.2.2" data-path="exposome.html"><a href="exposome.html#exposures-and-phenotypes-names"><i class="fa fa-check"></i><b>18.2.2</b> Exposures and phenotypes names</a></li>
<li class="chapter" data-level="18.2.3" data-path="exposome.html"><a href="exposome.html#summary-of-variables"><i class="fa fa-check"></i><b>18.2.3</b> Summary of variables</a></li>
<li class="chapter" data-level="18.2.4" data-path="exposome.html"><a href="exposome.html#missing-data"><i class="fa fa-check"></i><b>18.2.4</b> Missing data</a></li>
<li class="chapter" data-level="18.2.5" data-path="exposome.html"><a href="exposome.html#exposures-normality"><i class="fa fa-check"></i><b>18.2.5</b> Exposures Normality</a></li>
<li class="chapter" data-level="18.2.6" data-path="exposome.html"><a href="exposome.html#exposures-behaviour"><i class="fa fa-check"></i><b>18.2.6</b> Exposures Behaviour</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="exposome.html"><a href="exposome.html#exposures-imputation"><i class="fa fa-check"></i><b>18.3</b> Exposures Imputation</a></li>
<li class="chapter" data-level="18.4" data-path="exposome.html"><a href="exposome.html#performing-an-exwas"><i class="fa fa-check"></i><b>18.4</b> Performing an ExWAS</a></li>
<li class="chapter" data-level="18.5" data-path="exposome.html"><a href="exposome.html#exposures-pca"><i class="fa fa-check"></i><b>18.5</b> Exposures PCA</a></li>
<li class="chapter" data-level="18.6" data-path="exposome.html"><a href="exposome.html#exposures-correlation"><i class="fa fa-check"></i><b>18.6</b> Exposures Correlation</a></li>
</ul></li>
<li class="part"><span><b>V Developers</b></span></li>
<li class="chapter" data-level="19" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html"><i class="fa fa-check"></i><b>19</b> DSLite: DataSHIELD Implementation on Local Datasets</a>
<ul>
<li class="chapter" data-level="19.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#development-environment-setup"><i class="fa fa-check"></i><b>19.1</b> Development Environment Setup</a>
<ul>
<li class="chapter" data-level="19.1.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-packages"><i class="fa fa-check"></i><b>19.1.1</b> DataSHIELD Packages</a></li>
<li class="chapter" data-level="19.1.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#test-datasets"><i class="fa fa-check"></i><b>19.1.2</b> Test Datasets</a></li>
<li class="chapter" data-level="19.1.3" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#r-package-development-tools"><i class="fa fa-check"></i><b>19.1.3</b> R Package Development Tools</a></li>
</ul></li>
<li class="chapter" data-level="19.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-development-flow"><i class="fa fa-check"></i><b>19.2</b> DataSHIELD Development Flow</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#dsliteserver"><i class="fa fa-check"></i><b>19.2.1</b> DSLiteServer</a></li>
<li class="chapter" data-level="19.2.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-configuration"><i class="fa fa-check"></i><b>19.2.2</b> DataSHIELD Configuration</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#datashield-sessions"><i class="fa fa-check"></i><b>19.3</b> DataSHIELD Sessions</a></li>
<li class="chapter" data-level="19.4" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#debugging"><i class="fa fa-check"></i><b>19.4</b> Debugging</a></li>
<li class="chapter" data-level="19.5" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#limitations"><i class="fa fa-check"></i><b>19.5</b> Limitations</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#function-parameters-parser"><i class="fa fa-check"></i><b>19.5.1</b> Function Parameters Parser</a></li>
<li class="chapter" data-level="19.5.2" data-path="dslite-datashield-implementation-on-local-datasets.html"><a href="dslite-datashield-implementation-on-local-datasets.html#server-side-environments"><i class="fa fa-check"></i><b>19.5.2</b> Server Side Environments</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="package-info.html"><a href="package-info.html"><i class="fa fa-check"></i><b>20</b> Creating DataSHIELD packages</a>
<ul>
<li class="chapter" data-level="20.1" data-path="package-info.html"><a href="package-info.html#example-of-datashield-package-development"><i class="fa fa-check"></i><b>20.1</b> Example of DataSHIELD package development</a>
<ul>
<li class="chapter" data-level="20.1.1" data-path="package-info.html"><a href="package-info.html#fandfiles"><i class="fa fa-check"></i><b>20.1.1</b> Folders and files</a></li>
<li class="chapter" data-level="20.1.2" data-path="package-info.html"><a href="package-info.html#development-of-a-function"><i class="fa fa-check"></i><b>20.1.2</b> Development of a function</a></li>
<li class="chapter" data-level="20.1.3" data-path="package-info.html"><a href="package-info.html#testing-of-the-function"><i class="fa fa-check"></i><b>20.1.3</b> Testing of the function</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html"><i class="fa fa-check"></i><b>21</b> Tips and tricks</a>
<ul>
<li class="chapter" data-level="21.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-create-a-new-project-into-opal"><i class="fa fa-check"></i><b>21.1</b> How to create a new project into OPAL</a>
<ul>
<li class="chapter" data-level="21.1.1" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#manually"><i class="fa fa-check"></i><b>21.1.1</b> Manually</a></li>
<li class="chapter" data-level="21.1.2" data-path="extension-to-ga4gh-and-ega-bamvcf-files.html"><a href="extension-to-ga4gh-and-ega-bamvcf-files.html#using-r"><i class="fa fa-check"></i><b>21.1.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="21.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-upload-a-new-resource-into-opal"><i class="fa fa-check"></i><b>21.2</b> How to upload a new resource into OPAL</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#create_manually"><i class="fa fa-check"></i><b>21.2.1</b> Manually</a></li>
<li class="chapter" data-level="21.2.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#using-r-1"><i class="fa fa-check"></i><b>21.2.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-install-datashield-packages-into-opal-server"><i class="fa fa-check"></i><b>21.3</b> How to install DataSHIELD packages into OPAL server</a>
<ul>
<li class="chapter" data-level="21.3.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#manually-1"><i class="fa fa-check"></i><b>21.3.1</b> Manually</a></li>
<li class="chapter" data-level="21.3.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#using-r-2"><i class="fa fa-check"></i><b>21.3.2</b> Using R</a></li>
</ul></li>
<li class="chapter" data-level="21.4" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#how-to-install-r-packages-into-opal-server"><i class="fa fa-check"></i><b>21.4</b> How to install R packages into OPAL server</a>
<ul>
<li class="chapter" data-level="21.4.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#manually-2"><i class="fa fa-check"></i><b>21.4.1</b> Manually</a></li>
<li class="chapter" data-level="21.4.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#using-r-3"><i class="fa fa-check"></i><b>21.4.2</b> Using R</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="advancedtechnical-questions.html"><a href="advancedtechnical-questions.html"><i class="fa fa-check"></i><b>22</b> Advanced/technical questions</a></li>
<li class="part"><span><b>VI Appendix</b></span></li>
<li class="chapter" data-level="23" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>23</b> Session Info</a></li>
<li class="chapter" data-level="24" data-path="contributors.html"><a href="contributors.html"><i class="fa fa-check"></i><b>24</b> Contributors</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Orchestrating privacy-protected big data analyses of data from different resources with R and DataSHIELD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genome-wide-association-analyses-gwas" class="section level1" number="16">
<h1><span class="header-section-number">16</span> Genome-wide association analyses (GWAS)</h1>
<p>Genomic data can be stored in different formats. <a href="https://www.internationalgenome.org/wiki/Analysis/vcf4.0/">VCF</a> and <a href="http://zzz.bwh.harvard.edu/plink/">PLINK</a> files are commonly used in genetic epidemiology studies. We have a GWAS example available at <a href="https://github.com/isglobal-brge/brgedata">BRGE data repository</a> that aims to find SNPs associated with asthma. We have data stored in VCF (brge.vcf) with several covariates and phenotypes available in the file brge.txt (gender, age, obesity, smoking, country and asthma status). The same data is also available in PLINK format (brge.bed, brge.bim, brge.fam) with covariates in the file brge.phe.</p>
<p>Here we illustrate how to perform GWAS using R and Bioconductor packages or PLINK shell command line.</p>
<div id="gwasBioc" class="section level2" number="16.1">
<h2><span class="header-section-number">16.1</span> GWAS with Bioconductor</h2>
<p>We have created a resource having the VCF file of our study on asthma as previously described. The name of the resource is <code>brge_vcf</code> the phenotypes are available in another resource called <code>brge</code> that is a .txt file. The GWAS analysis is then perform as follows</p>
<p>We first start by preparing login data</p>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="genome-wide-association-analyses-gwas.html#cb204-1"></a>builder &lt;-<span class="st"> </span><span class="kw">newDSLoginBuilder</span>()</span>
<span id="cb204-2"><a href="genome-wide-association-analyses-gwas.html#cb204-2"></a>builder<span class="op">$</span><span class="kw">append</span>(<span class="dt">server =</span> <span class="st">&quot;study1&quot;</span>, <span class="dt">url =</span> <span class="st">&quot;https://opal-demo.obiba.org&quot;</span>,</span>
<span id="cb204-3"><a href="genome-wide-association-analyses-gwas.html#cb204-3"></a>               <span class="dt">user =</span> <span class="st">&quot;dsuser&quot;</span>, <span class="dt">password =</span> <span class="st">&quot;password&quot;</span>,</span>
<span id="cb204-4"><a href="genome-wide-association-analyses-gwas.html#cb204-4"></a>               <span class="dt">resource =</span> <span class="st">&quot;RSRC.brge_vcf&quot;</span>, <span class="dt">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>)</span>
<span id="cb204-5"><a href="genome-wide-association-analyses-gwas.html#cb204-5"></a>logindata &lt;-<span class="st"> </span>builder<span class="op">$</span><span class="kw">build</span>()</span>
<span id="cb204-6"><a href="genome-wide-association-analyses-gwas.html#cb204-6"></a></span>
<span id="cb204-7"><a href="genome-wide-association-analyses-gwas.html#cb204-7"></a>conns &lt;-<span class="st"> </span><span class="kw">datashield.login</span>(<span class="dt">logins =</span> logindata, <span class="dt">assign =</span> <span class="ot">TRUE</span>,</span>
<span id="cb204-8"><a href="genome-wide-association-analyses-gwas.html#cb204-8"></a>                          <span class="dt">symbol =</span> <span class="st">&quot;res&quot;</span>)</span></code></pre></div>
<p>In this case we have to assign to different resources. One for the VCF (obesity_vcf) and another one for the phenotypic data (obesity). To this end, the <code>datashield.assign.resource</code> function is required before assigning any object to the specific resource. Notice that the VCF resource can be load into R as a GDS thanks to <a href="#ext_VCF">our extension</a> of existing resources in the <em><a href="https://CRAN.R-project.org/package=reourcer">reourcer</a></em></p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="genome-wide-association-analyses-gwas.html#cb205-1"></a><span class="kw">datashield.assign.resource</span>(conns, <span class="dt">symbol =</span> <span class="st">&quot;vcf.res&quot;</span>, </span>
<span id="cb205-2"><a href="genome-wide-association-analyses-gwas.html#cb205-2"></a>                           <span class="dt">resource =</span> <span class="kw">list</span>(<span class="dt">study1 =</span> <span class="st">&quot;RSRC.brge_vcf&quot;</span>))</span>
<span id="cb205-3"><a href="genome-wide-association-analyses-gwas.html#cb205-3"></a><span class="kw">datashield.assign.expr</span>(conns, <span class="dt">symbol =</span> <span class="st">&quot;gds&quot;</span>, </span>
<span id="cb205-4"><a href="genome-wide-association-analyses-gwas.html#cb205-4"></a>                       <span class="dt">expr =</span> <span class="kw">quote</span>(<span class="kw">as.resource.object</span>(vcf.res)))</span>
<span id="cb205-5"><a href="genome-wide-association-analyses-gwas.html#cb205-5"></a></span>
<span id="cb205-6"><a href="genome-wide-association-analyses-gwas.html#cb205-6"></a></span>
<span id="cb205-7"><a href="genome-wide-association-analyses-gwas.html#cb205-7"></a><span class="kw">datashield.assign.resource</span>(conns, <span class="dt">symbol =</span> <span class="st">&quot;covars.res&quot;</span>, </span>
<span id="cb205-8"><a href="genome-wide-association-analyses-gwas.html#cb205-8"></a>                           <span class="dt">resource =</span> <span class="kw">list</span>(<span class="dt">study1 =</span> <span class="st">&quot;RSRC.brge&quot;</span>))</span>
<span id="cb205-9"><a href="genome-wide-association-analyses-gwas.html#cb205-9"></a><span class="kw">datashield.assign.expr</span>(conns, <span class="dt">symbol =</span> <span class="st">&quot;covars&quot;</span>, </span>
<span id="cb205-10"><a href="genome-wide-association-analyses-gwas.html#cb205-10"></a>                       <span class="dt">expr =</span> <span class="kw">quote</span>(<span class="kw">as.resource.data.frame</span>(covars.res)))</span></code></pre></div>
<p>These are the objects available in the Opal server</p>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="genome-wide-association-analyses-gwas.html#cb206-1"></a><span class="kw">ds.ls</span>()</span></code></pre></div>
<pre><code>$study1
$study1$environment.searched
[1] &quot;R_GlobalEnv&quot;

$study1$objects.found
[1] &quot;covars&quot;     &quot;covars.res&quot; &quot;gds&quot;        &quot;res&quot;        &quot;vcf.res&quot;   </code></pre>
<p>We can use <em><a href="https://github.com/datashield/dsBaseClient">dsBaseClient</a></em> functions to inspect the variables that are in the <code>covars</code> data.frame. The variables are</p>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="genome-wide-association-analyses-gwas.html#cb208-1"></a><span class="kw">ds.colnames</span>(<span class="st">&quot;covars&quot;</span>)</span></code></pre></div>
<pre><code>$study1
[1] &quot;scanID&quot;  &quot;gender&quot;  &quot;obese&quot;   &quot;age&quot;     &quot;smoke&quot;   &quot;country&quot; &quot;asthma&quot; </code></pre>
<p>The <code>asthma</code> variable has this number of individuals at each level (0: controls, 1: cases)</p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="genome-wide-association-analyses-gwas.html#cb210-1"></a><span class="kw">ds.table</span>(<span class="st">&quot;covars$asthma&quot;</span>)</span></code></pre></div>
<pre><code>
 Data in all studies were valid 

Study 1 :  No errors reported from this study</code></pre>
<pre><code>$output.list
$output.list$TABLE_rvar.by.study_row.props
             study
covars$asthma 1
            0 1
            1 1

$output.list$TABLE_rvar.by.study_col.props
             study
covars$asthma         1
            0 0.6864187
            1 0.3135813

$output.list$TABLE_rvar.by.study_counts
             study
covars$asthma    1
            0 1587
            1  725

$output.list$TABLES.COMBINED_all.sources_proportions
covars$asthma
    0     1 
0.686 0.314 

$output.list$TABLES.COMBINED_all.sources_counts
covars$asthma
   0    1 
1587  725 


$validity.message
[1] &quot;Data in all studies were valid&quot;</code></pre>
<p>There may be interest in only studying certain genes, for that matter, the loaded VCF resource can be subsetting as follows</p>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="genome-wide-association-analyses-gwas.html#cb213-1"></a>genes &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;A1BG&quot;</span>,<span class="st">&quot;A2MP1&quot;</span>)</span>
<span id="cb213-2"><a href="genome-wide-association-analyses-gwas.html#cb213-2"></a><span class="kw">ds.getSNPSbyGen</span>(<span class="st">&quot;gds&quot;</span>, genes)</span></code></pre></div>
<p>The previous code will over-write the VCF with the SNPs corresponding to the selected genes, if the intention is to perform studies with both the complete VCF and a subsetted one, the argument <code>name</code> can be used to create a new object on the server with the subsetted VCF, preserving the complete one.</p>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="genome-wide-association-analyses-gwas.html#cb214-1"></a>genes &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;A1BG&quot;</span>,<span class="st">&quot;A2MP1&quot;</span>)</span>
<span id="cb214-2"><a href="genome-wide-association-analyses-gwas.html#cb214-2"></a><span class="kw">ds.getSNPSbyGen</span>(<span class="st">&quot;gds&quot;</span>, <span class="dt">genes =</span> genes, <span class="dt">name =</span> <span class="st">&quot;subset.vcf&quot;</span>)</span></code></pre></div>
<p>Then, an object of class <code>GenotypeData</code> must be created at the server side to perform genetic data analyses. This is a container defined in the <em><a href="https://bioconductor.org/packages/3.11/GWASTools">GWASTools</a></em> package for storing genotype and phenotypic data from genetic association studies. By doing that we will also verify whether individuals in the GDS (e.g VCF) and covariates files have the same individuals and are in the same order. This can be performed by</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="genome-wide-association-analyses-gwas.html#cb215-1"></a><span class="kw">ds.GenotypeData</span>(<span class="dt">x=</span><span class="st">&#39;gds&#39;</span>, <span class="dt">covars =</span> <span class="st">&#39;covars&#39;</span>, <span class="dt">columnId =</span> <span class="dv">1</span>, <span class="dt">newobj.name =</span> <span class="st">&#39;gds.Data&#39;</span>)</span></code></pre></div>
<p>Before performing the association analyses, quality control (QC) can be performed to the loaded data. Three methodologies are available; 1) Principal Component Analysis (PCA) of the genomic data, 2) Hardy-Weinberg Equilibrium (HWE) testing and 3) Allelic frequency estimates. The QC methods 2 and 3 have as inputs a <em>GenotypeData</em> object, created with a covariates file that has a gender column; while method 1 has as input a <em>VCF</em>.</p>
<p>To perform the PCA, a pruning functionality is built inside so that redundant SNPs are discarted (there is an extra argument <code>ld.threshold</code> which controls the pruning, more information about it at the <a href="https://rdrr.io/bioc/SNPRelate/man/snpgdsLDpruning.html">SNPRelate documentation</a>), speeding up the execution time</p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="genome-wide-association-analyses-gwas.html#cb216-1"></a><span class="kw">ds.PCASNPS</span>(<span class="st">&quot;gds&quot;</span>, <span class="dt">prune =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="fig/PCASNPs-1.png" width="672" /></p>
<p>To perform QC methodologies 2 and 3, the name of the gender column as well as the keys to describe male or female have to be provided. Remember that we can visualize the names of the variables from our data by executing <code>ds.colnames("covars")</code>. In our case, this variable is called “gender”, and the levels of this variable are 1 for male and 2 for female as we can see here (NOTE: we cannot use <code>ds.levels</code> since gender variable is not a factor):</p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="genome-wide-association-analyses-gwas.html#cb217-1"></a><span class="kw">ds.table1D</span>(<span class="st">&quot;covars$gender&quot;</span>)<span class="op">$</span>counts</span></code></pre></div>
<pre><code>      covars$gender
1              1215
2              1097
Total          2312</code></pre>
<p>The HWE test can be performed to selected chromosomes using the argument <code>chromosomes</code>, only the autosomes can be selected when performing a HWE test, the encoding of the autosomes can be fetched with</p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="genome-wide-association-analyses-gwas.html#cb219-1"></a><span class="kw">ds.getChromosomeNames</span>(<span class="st">&quot;gds.Data&quot;</span>)<span class="op">$</span>autosomes</span></code></pre></div>
<pre><code>NULL</code></pre>
<p>Therefore, HWE can be performed by:</p>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="genome-wide-association-analyses-gwas.html#cb221-1"></a><span class="kw">ds.exactHWE</span>(<span class="st">&quot;gds.Data&quot;</span>, <span class="dt">sexcol =</span> <span class="st">&quot;gender&quot;</span>, <span class="dt">male =</span> <span class="st">&quot;1&quot;</span>, <span class="dt">female =</span> <span class="st">&quot;2&quot;</span>, <span class="dt">chromosome =</span> <span class="st">&quot;22&quot;</span>)</span></code></pre></div>
<pre><code>$study1
# A tibble: 1,581 x 9
   snpID chr   nAA   nAB   nBB   MAF                minor.allele f                   pval                
   &lt;int&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;              &lt;chr&gt;        &lt;chr&gt;               &lt;chr&gt;               
 1 95140 22    531   1082  640   0.475810031069685  A            0.0372494534103404  0.0762378420092437  
 2 95141 22    580   1092  590   0.497789566755084  A            0.0344638881244839  0.101115839682148   
 3 95142 22    0     333   1951  0.0728984238178634 A            -0.0786304604486423 3.29321452000707e-06
 4 95143 22    0     225   2080  0.0488069414316703 A            -0.0513112884834663 0.00571353422173819 
 5 95144 22    176   831   1249  0.262189716312057  A            0.0479240933754878  0.025554465064758   
 6 95145 22    249   1003  1055  0.32531426094495   A            0.00958157673233362 0.635607524466151   
 7 95146 22    55    516   1738  0.135556517973149  A            0.0464603328061876  0.0325126363699498  
 8 95147 22    203   929   1164  0.290722996515679  A            0.0188880417746163  0.362631409282596   
 9 95148 22    291   1035  985   0.349848550411077  A            0.0154998317584483  0.464409674690009   
10 95149 22    11    276   2013  0.0647826086956522 A            0.00966929694008412 0.604847402951204   
# ... with 1,571 more rows

attr(,&quot;class&quot;)
[1] &quot;dsexactHWE&quot; &quot;list&quot;      </code></pre>
<p>Similarly, allele frequencies estimates can be estimated by:</p>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="genome-wide-association-analyses-gwas.html#cb223-1"></a><span class="kw">ds.alleleFrequency</span>(<span class="st">&quot;gds.Data&quot;</span>, <span class="dt">sexcol =</span> <span class="st">&quot;gender&quot;</span>, <span class="dt">male =</span> <span class="st">&quot;1&quot;</span>, <span class="dt">female =</span> <span class="st">&quot;2&quot;</span>)</span></code></pre></div>
<pre><code>$study1
# A tibble: 99,289 x 7
          M       F      all   n.M   n.F     n      MAF
      &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;
 1 0.015    0.0147  0.0149    1200  1086  2286 0.0149  
 2 0.000837 0.00276 0.00175   1195  1087  2282 0.00175 
 3 0.00330  0.00548 0.00433   1212  1095  2307 0.00433 
 4 0.0124   0.00868 0.0106    1213  1095  2308 0.0106  
 5 0.00165  0       0.000866  1214  1095  2309 0.000866
 6 0.0712   0.0754  0.0732    1208  1094  2302 0.0732  
 7 0.00660  0.00913 0.00780   1213  1095  2308 0.00780 
 8 0.00248  0.00274 0.00261   1209  1093  2302 0.00261 
 9 0.182    0.181   0.181     1188  1071  2259 0.181   
10 0.00165  0       0.000868  1210  1094  2304 0.000868
# ... with 99,279 more rows

attr(,&quot;class&quot;)
[1] &quot;dsalleleFrequency&quot; &quot;list&quot;             </code></pre>
<p>In the future, more functions will be created to perform quality control (QC) for both, SNPs and inviduals.</p>
<p>Association analysis for a given SNP is performed by simply</p>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="genome-wide-association-analyses-gwas.html#cb225-1"></a><span class="kw">ds.glmSNP</span>(<span class="dt">snps.fit =</span> <span class="st">&quot;rs11247693&quot;</span>, <span class="dt">model =</span> asthma <span class="op">~</span><span class="st"> </span>gender <span class="op">+</span><span class="st"> </span>age, <span class="dt">genoData=</span><span class="st">&#39;gds.Data&#39;</span>)</span></code></pre></div>
<pre><code>             Estimate Std. Error   p-value
rs11247693 -0.1543215  0.2309585 0.5040196
attr(,&quot;class&quot;)
[1] &quot;dsGlmSNP&quot; &quot;matrix&quot;   &quot;array&quot;   </code></pre>
<p>The analysis of all available SNPs is performed when the argument <code>snps.fit</code> is missing. The function performs the analysis of the selected SNPs in a single repository or in multiple repositories as performing pooled analyses (it uses <code>ds.glm</code> DataSHIELD function). As in the case of transcriptomic data, analyzing all the SNPs in the genome (e.g GWAS) will be high time-consuming. We can adopt a similar approach as the one adopted using the <em><a href="https://bioconductor.org/packages/3.11/limma">limma</a></em> at each server. That is, we run GWAS at each repository using specific and scalable packages available in R/Bioc. In that case we use the <em><a href="https://bioconductor.org/packages/3.11/GWASTools">GWASTools</a></em> and <em><a href="https://bioconductor.org/packages/3.11/GENESIS">GENESIS</a></em> packages. The complete pipeline is implemented in this function</p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="genome-wide-association-analyses-gwas.html#cb227-1"></a>ans.bioC &lt;-<span class="st"> </span><span class="kw">ds.GWAS</span>(<span class="st">&#39;gds.Data&#39;</span>, <span class="dt">model=</span>asthma<span class="op">~</span>age<span class="op">+</span>country)</span></code></pre></div>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="genome-wide-association-analyses-gwas.html#cb228-1"></a>ans.bioC</span></code></pre></div>
<pre><code>$study1
# A tibble: 99,288 x 14
   variant.id rs         chr         pos n.obs    freq   MAC Score Score.SE Score.Stat  Score.pval    Est Est.SE     PVE
        &lt;int&gt; &lt;chr&gt;      &lt;chr&gt;     &lt;int&gt; &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;      &lt;dbl&gt;       &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;
 1      97800 rs12008773 X      64955287  2227 0.0126     56  19.5     3.84       5.07 0.000000408  1.32  0.260  0.0111 
 2      91449 rs2267914  20      1858988  2311 0.104     481  47.5     9.62       4.93 0.000000809  0.513 0.104  0.0105 
 3      19742 rs7153     3     122865987  2308 0.256    1181  62.3    13.7        4.56 0.00000508   0.334 0.0732 0.00900
 4      93268 rs6097326  20     51324797  2311 0.125     579  46.4    10.3        4.51 0.00000642   0.439 0.0973 0.00881
 5      19744 rs3732410  3     122898410  2308 0.254    1173  60.3    13.6        4.43 0.00000940   0.325 0.0734 0.00849
 6      66984 rs11055608 12     13804693  2306 0.446    2057 -67.6    15.7       -4.31 0.0000161   -0.275 0.0638 0.00805
 7      74760 rs7995146  13    111804035  2304 0.0527    243  29.7     6.96       4.27 0.0000195    0.614 0.144  0.00789
 8      59678 rs7098143  10     83329037  2295 0.210     963 -54.5    12.8       -4.25 0.0000214   -0.331 0.0780 0.00781
 9      13835 rs13002717 2     186197782  2311 0.00649    30  10.6     2.52       4.22 0.0000244    1.67  0.397  0.00770
10      27337 rs1602679  4     167576714  2299 0.101     463  39.7     9.46       4.20 0.0000264    0.444 0.106  0.00764
# ... with 99,278 more rows

attr(,&quot;class&quot;)
[1] &quot;dsGWAS&quot; &quot;list&quot;  </code></pre>
<p>This close the DataSHIELD session</p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="genome-wide-association-analyses-gwas.html#cb230-1"></a><span class="kw">datashield.logout</span>(conns)</span></code></pre></div>
</div>
<div id="gwas-with-plink" class="section level2" number="16.2">
<h2><span class="header-section-number">16.2</span> GWAS with PLINK</h2>
<p>Here we illustrate how to perform the same GWAS analyses on the asthma using PLINK secure shell commands. This can be performed thanks to the posibility of having ssh resources as described <a href="shell-extension.html#shell_extension">here</a>.</p>
<p>It is worth to notice that this workflow and the new R functions implemented in <em><a href="https://github.com/isglobal-brge/dsOmicsClient">dsOmicsClient</a></em> could be used as a guideline to carry out similar analyses using existing analysis tools in genomics such as IMPUTE, SAMtools or BEDtools among many others.</p>
<p>We start by assigning login resources</p>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="genome-wide-association-analyses-gwas.html#cb231-1"></a>  <span class="kw">library</span>(DSOpal)</span>
<span id="cb231-2"><a href="genome-wide-association-analyses-gwas.html#cb231-2"></a>  <span class="kw">library</span>(dsBaseClient)</span>
<span id="cb231-3"><a href="genome-wide-association-analyses-gwas.html#cb231-3"></a>  <span class="kw">library</span>(dsOmicsClient)</span>
<span id="cb231-4"><a href="genome-wide-association-analyses-gwas.html#cb231-4"></a>  builder &lt;-<span class="st"> </span><span class="kw">newDSLoginBuilder</span>()</span>
<span id="cb231-5"><a href="genome-wide-association-analyses-gwas.html#cb231-5"></a>  builder<span class="op">$</span><span class="kw">append</span>(<span class="dt">server =</span> <span class="st">&quot;study1&quot;</span>, <span class="dt">url =</span> <span class="st">&quot;https://opal-demo.obiba.org&quot;</span>,</span>
<span id="cb231-6"><a href="genome-wide-association-analyses-gwas.html#cb231-6"></a>                 <span class="dt">user =</span> <span class="st">&quot;dsuser&quot;</span>, <span class="dt">password =</span> <span class="st">&quot;password&quot;</span>,</span>
<span id="cb231-7"><a href="genome-wide-association-analyses-gwas.html#cb231-7"></a>                 <span class="dt">resource =</span> <span class="st">&quot;RSRC.brge_plink&quot;</span>, <span class="dt">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>)</span>
<span id="cb231-8"><a href="genome-wide-association-analyses-gwas.html#cb231-8"></a>  logindata &lt;-<span class="st"> </span>builder<span class="op">$</span><span class="kw">build</span>()</span></code></pre></div>
<p>Then we assign the resource to a symbol (i.e. R object) called <code>client</code> which is a ssh resource</p>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="genome-wide-association-analyses-gwas.html#cb232-1"></a>  conns &lt;-<span class="st"> </span><span class="kw">datashield.login</span>(<span class="dt">logins =</span> logindata, <span class="dt">assign =</span> <span class="ot">TRUE</span>,</span>
<span id="cb232-2"><a href="genome-wide-association-analyses-gwas.html#cb232-2"></a>                            <span class="dt">symbol =</span> <span class="st">&quot;client&quot;</span>)</span>
<span id="cb232-3"><a href="genome-wide-association-analyses-gwas.html#cb232-3"></a>  <span class="kw">ds.class</span>(<span class="st">&quot;client&quot;</span>)</span></code></pre></div>
<pre><code>$study1
[1] &quot;SshResourceClient&quot;     &quot;CommandResourceClient&quot; &quot;ResourceClient&quot;        &quot;R6&quot;                   </code></pre>
<p>Now, we are ready to run any PLINK command from the client site. Notice that in this case we want to assess association between the genotype data in bed format and use as phenotype the variable ‘asthma’ that is in the file ‘brge.phe’ in the 6th column. The sentence in a PLINK command would be (NOTE: we avoid –out to indicate the output file since the file will be available in R as a tibble).</p>
<pre><code>  plink --bfile brge --logistic --pheno brge.phe --mpheno 6 --covar brge.phe --covar-name gender,age </code></pre>
<p>The arguments must be encapsulated in a single character without the command ‘plink’</p>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="genome-wide-association-analyses-gwas.html#cb235-1"></a>  plink.arguments &lt;-<span class="st"> &quot;--bfile brge --logistic --pheno brge.phe --mpheno 6 --covar brge.phe --covar-name gender,age&quot;</span></span></code></pre></div>
<p>the analyses are then performed by</p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="genome-wide-association-analyses-gwas.html#cb236-1"></a>  ans.plink &lt;-<span class="st"> </span><span class="kw">ds.PLINK</span>(<span class="st">&quot;client&quot;</span>, plink.arguments)</span></code></pre></div>
<p>The object <code>ans.plink</code> contains the PLINK results at each server as well as the outuput provided by PLINK</p>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="genome-wide-association-analyses-gwas.html#cb237-1"></a><span class="kw">lapply</span>(ans.plink, names)</span></code></pre></div>
<pre><code>$study1
[1] &quot;results&quot;   &quot;plink.out&quot;</code></pre>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="genome-wide-association-analyses-gwas.html#cb239-1"></a><span class="kw">head</span>(ans.plink<span class="op">$</span>study1<span class="op">$</span>results)</span></code></pre></div>
<pre><code># A tibble: 6 x 9
    CHR SNP           BP A1    TEST   NMISS    OR   STAT        P
  &lt;dbl&gt; &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;
1     0 MitoC3993T  3993 T     ADD     2286 0.752 -1.33  0.182   
2     0 MitoC3993T  3993 T     gender  2286 0.742 -3.27  0.00107 
3     0 MitoC3993T  3993 T     age     2286 1.00   0.565 0.572   
4     0 MitoG4821A  4821 A     ADD     2282 2.68   1.71  0.0879  
5     0 MitoG4821A  4821 A     gender  2282 0.740 -3.31  0.000940
6     0 MitoG4821A  4821 A     age     2282 1.00   0.465 0.642   </code></pre>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="genome-wide-association-analyses-gwas.html#cb241-1"></a>ans.plink<span class="op">$</span>study<span class="op">$</span>plink.out</span></code></pre></div>
<pre><code>$status
[1] 0

$output
 [1] &quot;&quot;                                                                                   
 [2] &quot;@----------------------------------------------------------@&quot;                       
 [3] &quot;|        PLINK!       |     v1.07      |   10/Aug/2009     |&quot;                       
 [4] &quot;|----------------------------------------------------------|&quot;                       
 [5] &quot;|  (C) 2009 Shaun Purcell, GNU General Public License, v2  |&quot;                       
 [6] &quot;|----------------------------------------------------------|&quot;                       
 [7] &quot;|  For documentation, citation &amp; bug-report instructions:  |&quot;                       
 [8] &quot;|        http://pngu.mgh.harvard.edu/purcell/plink/        |&quot;                       
 [9] &quot;@----------------------------------------------------------@&quot;                       
[10] &quot;&quot;                                                                                   
[11] &quot;Skipping web check... [ --noweb ] &quot;                                                 
[12] &quot;Writing this text to log file [ /tmp/ssh-2566/out.log ]&quot;                            
[13] &quot;Analysis started: Tue Jan 12 14:50:23 2021&quot;                                         
[14] &quot;&quot;                                                                                   
[15] &quot;Options in effect:&quot;                                                                 
[16] &quot;\t--bfile brge&quot;                                                                      
[17] &quot;\t--logistic&quot;                                                                        
[18] &quot;\t--covar brge.phe&quot;                                                                  
[19] &quot;\t--covar-name gender,age&quot;                                                           
[20] &quot;\t--noweb&quot;                                                                           
[21] &quot;\t--out /tmp/ssh-2566/out&quot;                                                           
[22] &quot;&quot;                                                                                   
[23] &quot;Reading map (extended format) from [ brge.bim ] &quot;                                   
[24] &quot;100000 markers to be included from [ brge.bim ]&quot;                                    
[25] &quot;Reading pedigree information from [ brge.fam ] &quot;                                    
[26] &quot;2312 individuals read from [ brge.fam ] &quot;                                           
[27] &quot;2312 individuals with nonmissing phenotypes&quot;                                        
[28] &quot;Assuming a disease phenotype (1=unaff, 2=aff, 0=miss)&quot;                              
[29] &quot;Missing phenotype value is also -9&quot;                                                 
[30] &quot;725 cases, 1587 controls and 0 missing&quot;                                             
[31] &quot;1097 males, 1215 females, and 0 of unspecified sex&quot;                                 
[32] &quot;Reading genotype bitfile from [ brge.bed ] &quot;                                        
[33] &quot;Detected that binary PED file is v1.00 SNP-major mode&quot;                              
[34] &quot;Reading 6 covariates from [ brge.phe ] with nonmissing values for 2199 individuals&quot; 
[35] &quot;Selected subset of 2 from 6 covariates&quot;                                             
[36] &quot;For these, nonmissing covariate values for 2312 individuals&quot;                        
[37] &quot;Before frequency and genotyping pruning, there are 100000 SNPs&quot;                     
[38] &quot;2312 founders and 0 non-founders found&quot;                                             
[39] &quot;6009 heterozygous haploid genotypes; set to missing&quot;                                
[40] &quot;Writing list of heterozygous haploid genotypes to [ /tmp/ssh-2566/out.hh ]&quot;         
[41] &quot;7 SNPs with no founder genotypes observed&quot;                                          
[42] &quot;Warning, MAF set to 0 for these SNPs (see --nonfounders)&quot;                           
[43] &quot;Writing list of these SNPs to [ /tmp/ssh-2566/out.nof ]&quot;                            
[44] &quot;Total genotyping rate in remaining individuals is 0.994408&quot;                         
[45] &quot;0 SNPs failed missingness test ( GENO &gt; 1 )&quot;                                        
[46] &quot;0 SNPs failed frequency test ( MAF &lt; 0 )&quot;                                           
[47] &quot;After frequency and genotyping pruning, there are 100000 SNPs&quot;                      
[48] &quot;After filtering, 725 cases, 1587 controls and 0 missing&quot;                            
[49] &quot;After filtering, 1097 males, 1215 females, and 0 of unspecified sex&quot;                
[50] &quot;Converting data to Individual-major format&quot;                                         
[51] &quot;Writing logistic model association results to [ /tmp/ssh-2566/out.assoc.logistic ] &quot;
[52] &quot;&quot;                                                                                   
[53] &quot;Analysis finished: Tue Jan 12 14:52:20 2021&quot;                                        
[54] &quot;&quot;                                                                                   

$error
character(0)

$command
[1] &quot;cd /home/master/brge &amp;&amp; plink1 --bfile brge --logistic --covar brge.phe --covar-name gender,age --noweb --out /tmp/ssh-2566/out&quot;

attr(,&quot;class&quot;)
[1] &quot;resource.exec&quot;</code></pre>
<p>We can compare the p-values obtained using PLINK with Bioconductor-based packages for the top-10 SNPs as follows:</p>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="genome-wide-association-analyses-gwas.html#cb243-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb243-2"><a href="genome-wide-association-analyses-gwas.html#cb243-2"></a><span class="co"># get SNP p.values (additive model - ADD)</span></span>
<span id="cb243-3"><a href="genome-wide-association-analyses-gwas.html#cb243-3"></a>res.plink &lt;-<span class="st"> </span>ans.plink<span class="op">$</span>study1<span class="op">$</span>results <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(TEST<span class="op">==</span><span class="st">&quot;ADD&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb243-4"><a href="genome-wide-association-analyses-gwas.html#cb243-4"></a><span class="kw">arrange</span>(P)</span>
<span id="cb243-5"><a href="genome-wide-association-analyses-gwas.html#cb243-5"></a><span class="co"># compare top-10 with Biocoductor&#39;s results</span></span>
<span id="cb243-6"><a href="genome-wide-association-analyses-gwas.html#cb243-6"></a>snps &lt;-<span class="st"> </span>res.plink<span class="op">$</span>SNP[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</span>
<span id="cb243-7"><a href="genome-wide-association-analyses-gwas.html#cb243-7"></a>plink &lt;-<span class="st"> </span>res.plink <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(SNP<span class="op">%in%</span>snps) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(SNP, P)</span>
<span id="cb243-8"><a href="genome-wide-association-analyses-gwas.html#cb243-8"></a>bioC &lt;-<span class="st"> </span>ans.bioC<span class="op">$</span>study1 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(rs<span class="op">%in%</span>snps) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(rs, Score.pval)</span>
<span id="cb243-9"><a href="genome-wide-association-analyses-gwas.html#cb243-9"></a><span class="kw">left_join</span>(plink, bioC, <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;SNP&quot;</span> =<span class="st"> &quot;rs&quot;</span>))</span></code></pre></div>
<pre><code># A tibble: 10 x 3
   SNP                 P  Score.pval
   &lt;chr&gt;           &lt;dbl&gt;       &lt;dbl&gt;
 1 rs2267914  0.00000151 0.000000809
 2 rs6097326  0.00000424 0.00000642 
 3 rs7153     0.00000440 0.00000508 
 4 rs3732410  0.00000817 0.00000940 
 5 rs7995146  0.0000170  0.0000195  
 6 rs6495788  0.0000213  0.0000278  
 7 rs1602679  0.0000268  0.0000264  
 8 rs11055608 0.0000270  0.0000161  
 9 rs7098143  0.0000313  0.0000214  
10 rs7676164  0.0000543  0.0000537  </code></pre>
<p>As expected, the p-values are in the same order of magnitud having little variations due to the implemented methods of each software.</p>
<p>We can do the same comparions of minor allele frequency (MAF) estimation performed with Bioconductor and PLINK. To this end, we need first to estimate MAF using PLINK</p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="genome-wide-association-analyses-gwas.html#cb245-1"></a>  plink.arguments &lt;-<span class="st"> &quot;--bfile brge --freq&quot;</span></span>
<span id="cb245-2"><a href="genome-wide-association-analyses-gwas.html#cb245-2"></a>  ans.plink2 &lt;-<span class="st"> </span><span class="kw">ds.PLINK</span>(<span class="st">&quot;client&quot;</span>, plink.arguments)</span>
<span id="cb245-3"><a href="genome-wide-association-analyses-gwas.html#cb245-3"></a>  maf.plink &lt;-<span class="st"> </span>ans.plink2<span class="op">$</span>study1<span class="op">$</span>results</span>
<span id="cb245-4"><a href="genome-wide-association-analyses-gwas.html#cb245-4"></a>  </span>
<span id="cb245-5"><a href="genome-wide-association-analyses-gwas.html#cb245-5"></a>  plink &lt;-<span class="st"> </span>maf.plink <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(SNP<span class="op">%in%</span>snps) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(SNP, MAF)</span>
<span id="cb245-6"><a href="genome-wide-association-analyses-gwas.html#cb245-6"></a>  bioC &lt;-<span class="st"> </span>ans.bioC<span class="op">$</span>study1 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(rs<span class="op">%in%</span>snps) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(rs, freq)</span>
<span id="cb245-7"><a href="genome-wide-association-analyses-gwas.html#cb245-7"></a>  <span class="kw">left_join</span>(plink, bioC, <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;SNP&quot;</span> =<span class="st"> &quot;rs&quot;</span>))</span></code></pre></div>
<pre><code># A tibble: 10 x 3
   SNP           MAF   freq
   &lt;chr&gt;       &lt;dbl&gt;  &lt;dbl&gt;
 1 rs7153     0.256  0.256 
 2 rs3732410  0.254  0.254 
 3 rs7676164  0.304  0.304 
 4 rs1602679  0.101  0.101 
 5 rs7098143  0.210  0.210 
 6 rs11055608 0.446  0.446 
 7 rs7995146  0.0527 0.0527
 8 rs6495788  0.267  0.267 
 9 rs2267914  0.104  0.104 
10 rs6097326  0.125  0.125 </code></pre>
<p>This close the DataSHIELD session</p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="genome-wide-association-analyses-gwas.html#cb247-1"></a>  <span class="kw">datashield.logout</span>(conns)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="epigenome-wide-association-analysis-ewas.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="GIS.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
